%!TEX root = ../plos_template.tex
Different gene regulatory network architectures permit different collections of gene expression patterns, defined by the time-dependent expression states of all genes under consideration, which ultimately determine phenotypes~\cite{Alon2007,Milo2004}. The impact of stochasticity in gene expression~\cite{Eldar2010,Sanchez2013} upon the relationship between each network architecture and its associated potential expression pattern repertoire is not clear~\cite{Jothi2009,Golding2005,Lestas2010,Hilfinger2012,Chalancon2012}. In order to investigate this relationship, we take as fundamental the lowest level genotype-phenotype mapping: that from gene regulatory networks to gene expression patterns. Mathematical formalization of these mappings results in the association of a space of probability distributions to each gene regulatory network architecture that defines the latter's functional repertoire. Here we show that networks possessing cycles~\cite{Wainwright2007} in the gene regulatory network topology, and thereby \emph{lacking} certain kinds of higher-order correlations among genes, are incapable of satisfying functional requirements that may be imposed upon them by natural selection. This insight derives from a quantitative comparison of the geometries of spaces of probability distributions on genotype-phenotype maps for different network architectures. These results and the associated framework highlight the context-dependence of the functionality of stochastic gene regulatory networks and may also conribute to an explanation for the observation of hierarchical modular gene regulatory network architectures throughout the tree of life~\cite{Ravasz2002,Segre2005,Wagner2007,Erwin2009,Jothi2009,Bhardwaj2010,Colm}.

% permit natural selection to act upon expression patterns that are unavailable to network topologies lacking these features.
% The function associated to a given gene regulatory network architecture is a context-dependent feature of genetic networks in general.
% That the \emph{restriction} of stochastic genetic interactions can enable rather than limit access to certain selectable phenotypes may help explain why hierarchical modular gene regulatory network architectures are observed throughout the tree of life.

% Different gene regulatory network architectures are commonly thought to provide access to different collections of gene expression patterns that may ultimately result in different phenotypes~\cite{Alon2007}. The manner in which stochasticity in the gene expression process~\cite{Eldar2010,Sanchez2013} impacts the relationship between network architecture and the gene expression patterns each is capable of producing is not clear~\cite{Jothi2009,Chalancon2012}. In order to investigate this relationship, we work with the lowest level genotype-phenotype mapping, from genotypes to gene expression patterns, and adapt a formalism to reason generally about probability distributions over such maps~\cite{Lane1998,MacLane1992,Awodey2006,Abramsky2011}. In this formalism, network architectures are represented as hypergraphs that can additionally be used to specify the form of a hierarchical model~\cite{Lauritzen1996} for the stationary probability distribution of a stochastic process modelling gene expression in a manner that takes into account interactions among genes. In order to compare the accessible gene expression patterns, we compare the geometries of the entire space of probability distributions on genotype-phenotype maps for each network architecure. We show that under certain conditions on the network context, and paradoxically from one point of view, those involving restrictions from higher-order to lower-order interactions permit access to additional gene expression patterns. This relationship holds when the network architecture resulting from the restriction process contains a subnetwork forming a cycle. That these additional gene expression patterns may enable the fulfillment of functional requirements in certain environments, while remaining intrinsically inaccessible when the highest-order interactions relative to a given number of genes are allowed, implies that, even when gene expression stochasticity is taken into account, network architectures may be functionally differentiated and thus serve as substrate for natural selection in a background of otherwise evolutionarily neutral space. The recognition that the \emph{restriction} of interactions~\cite{Bar-Even2006,Johnson2010a} can be an enabler as opposed to a limitation on accessible phenotypes in certain network contexts may contribute to a more general explanation for the hierarchical modular architecture~\cite{Ravasz2002,Segre2005,Wagner2007,Erwin2009,Jothi2009,Bhardwaj2010,Colm} of the gene regulatory networks observed throughout the tree of life.

% We show in a simple case where the highest-order gene regulatory interactions are allowed, that under some conditions the collection of accessible phenotypes is smaller than in some of the alternative cases in which only lower-order interactions are allowed. This conclusion is arrived at by comparing the geometries of spaces of probability distributions on genotype-phenotype maps for the case of a genome containing a given number of simultaneously interacting genes to the equivalent for all of the other possible gene regulatory network topologies on the same number of genes. We find that the latter spaces of probability distributions associated to the restriction to lower-order interactions are actually larger than the former one
%Comparing the geometries of spaces of probability distributions on genotype-phenotype maps, we show that those involving restrictions from higher-order to lower-order interactions in such a manner that the network topology contains a subnetwork forming a cycle permit access to additional correlation patterns of expression among multiple interacting genes.
