%!TEX root = ../plos_template.tex
Different gene regulatory network architectures are commonly thought to provide access to different collections of gene expression patterns that may ultimately result in different phenotypes~\cite{Alon2007}. The manner in which stochasticity in the gene expression process~\cite{Eldar2010,Sanchez2013} impacts the relationship between network architecture and the expression patterns each is capable of producing is not clear~\cite{Jothi2009,Chalancon2012}. In order to investigate this relationship, we begin with the finest grained notion of genotype to phenotype map, gene expression, and adopt a formalism to reason generally about probability distributions over such maps~\cite{Lane1998,MacLane1992,Awodey2006,Abramsky2011}. In all cases, this must be relative to a notion of network architecture, defined here as a hypergraph determining a hierarchical model~\cite{Lauritzen1996} over a given set of genes. We show that in a simple case where the highest-order gene regulatory interactions are allowed, then under some conditions the collection of accessible phenotypes is smaller than in some of the alternative cases in which only lower-order interactions are allowed. This conclusion is arrived at by comparing the geometries of spaces of probability distributions on genotype-phenotype maps for the case of a genome containing a given number of simultaneously interacting genes to the equivalent for all of the other possible gene regulatory network topologies on the same number of genes. We find that the latter spaces of probability distributions associated to the restriction to lower-order interactions are actually larger than the former one whenever the gene regulatory network topology contains a cyclic structure. A generalization of this result to an arbitrary number of genes implies that modularization of interactions, defined here as restriction from relatively higher- to lower-order interactions, within and among gene regulatory networks permits access to additional correlation patterns of expression among multiple interacting genes. That these additional gene expression patterns may enable the fulfillment of functional requirements in certain environments while remaining intrinsically inaccessible when highest-order interactions are allowed implies that, even when gene expression stochasticity is taken into account, network architectures may be functionally differentiated and thus serve as substrate for natural selection in a background of otherwise evolutionarily neutral space. The identification of \emph{interaction constraint}~\cite{Bar-Even2006,Johnson2010a} as an enabler as opposed to a limitation on accessible phenotypes may contribute to a more general explanation for the hierarchical modular architecture~\cite{Ravasz2002,Segre2005,Wagner2007,Erwin2009,Jothi2009,Bhardwaj2010,Colm} of the gene regulatory networks observed throughout the tree of life.
