%!TEX root = ../plos_template.tex
It is intuitive that a more complex gene regulatory network architecture may provide access to a higher diversity of phenotypes~\cite{Adami2000,Carroll2001,Doebeli2010,Prochnik2010}. This explanation is suggested to justify the evolution of complex organisms, as a broader range of phenotypes may obtain higher fitness in a wider range of environments. Beginning with the highest resolution genotype to phenotype map, which we take to be gene expression, and using a formalism to reason generally about such maps~\cite{Lane1998,MacLane1992,Awodey2006,Abramsky2011}, we show that in a simple case where all possible gene regulatory interactions are allowed, the range of potentially accessible phenotypes is, relatively, constrained. This conclusion is arrived at by directly comparing the volumes of the spaces of consistent probability distributions on genotype-phenotype maps for the case of a genome containing four simultaneously interacting genes to otherwise identical genomes constrained to interact in any of the remaining potential gene regulatory network topologies. We find that the latter spaces of probability distributions are larger than the former whenever the gene regulatory network topology contains a cyclic structure. A generalization of this result to an arbitrary number of genes implies that modularization of interactions within and among gene regulatory networks is a requirement to access additional correlations among the expression of multiple interacting genes that serve as functional requirements in certain environments while remaining intrinsically inaccessible when all possible interactions are allowed. The identification of \emph{regulatory constraint}~\cite{Bar-Even2006,Johnson2010a} as a necessary enabler, as opposed to an inhibitor, of \emph{phenotypic diversity} may contribute to a more general explanation for the hierarchical modular architecture~\cite{Ravasz2002,Segre2005,Wagner2007,Erwin2009,Jothi2009,Bhardwaj2010,Colm} of the gene regulatory networks and, more generally, interactomes observed throughout the tree of life.
