%!TEX root = ../plos_template.tex
To contribute to the broader goal of establishing an integrated framework that synthesizes hypothesized intrinsic and extrinsic constraints necessary to understand the functioning and evolution of biological systems, here we have attempted to trace a path from relationships among spaces of probability distributions to their impact upon stochastic models of gene regulatory networks and, via the impact of gene expression on phenotypes, to evolutionary processes. One goal of measuring gene expression at transcriptomic scale has to be to uncover the structure of the generative process encoded in the GRN interactions involved, but, so far, even the most sophisticated methods of describing them at the mechanistic level are only solvable for extremely simple regulatory network architectures~\cite{Walczak2009,Mugler2009}. This fact has, in part, motivated computational biologists to develop a large collection of algorithms to infer aspects of this structure~\cite{DeSmet2010} and experimental biologists to compare networks on the basis of their hierarchical and modular architecture~\cite{Ideker2012}. Our model and its framework put forward a class of fundamental constraints, that while not unique to gene regulatory networks, must nevertheless impact their structure at the level of abstraction described here. The fact that GRNA impacts the structure of the entire space of probability distributions over genotype-phenotype maps---and not just a parameter-dependent fragment of it---accessible to each given architecture provides a clear mechanism by which natural selection may mold GRNAs.

It is clear that in most cases, when gene regulatory networks are studied, we remove a subnetwork from a larger context~\cite{Alon2007}. In our case, the network context, such as that schematized in \ref{fig:stochdynscheme}A, must play a crucial role, in addition to the subnetwork architecture upon which we have focused here, in determining whether or not it is possible to access the gene expression patterns that more highly modular regulatory network architectures ostensibly provide access to. One mechanism by which the network context may determine access to correlations that lie outside what is achievable in the non-modular case is via epigenetic regulation that relieves statistical dependence upon highest-order interactions implicit to the non-modular architectures we have described. On the other hand, some forms of epigenetic regulation may serve the opposite purpose to couple genes which do not otherwise interact. It will be important in future work to address these questions in the context of developing bottom-up stochastic process models that allow for the explicit encoding and solution of more complex gene regulatory networks~\cite{Walczak2009,Mugler2009}, which also allow for the incorporation of the effects of epigenetic regulation. Our results suggest that when requirements such as that in \ref{fig:inconsistentthreecycle}A are placed upon a network, then the network context must ultimately serve to decouple statistical dependency intrinsic to the network architecture, and this essentially requires the type of hierarchical relationship implicit in that among genetic and epigenetic regulation but also to that within hierarchically organized transcription factor networks \cite{Jothi2009,Bhardwaj2010,Chalancon2012,Colm}.

In biology, as in other sciences such as physics~\cite{Gell-Mann1956}, prevailing wisdom suggests that anything is allowed unless it has been demonstrated to some reasonable degree of confidence to be forbidden. We suspect that there are many other simple physical and mathematical constraints that can be brought to bear on biological problems. Constraint-based models that extract invariants of metabolic and other biochemical reaction networks provide a shining example of this approach~\cite{Karp2012,Bordbar2014}. The use of fundamental constraints to characterize geometric properties of phenotype space have begun to elicit applications of this approach to evolutionary processes~\cite{Shoval2012,Sheftel2013,Jordan2013}. At first, statements of such may only be effective at a relatively high level of abstraction. However, as a larger number of broad constraints are organized and integrated into a common language they can be applied to increasingly detailed models that integrate omic scale information over developmental and evolutionary timescales~\cite{Gunawardena2013}. Approaches of the type we have exhibited here hold promise for cashing out the tremendous potential that continues to be built up in databases of high-throughput genomic data.

The ultimate goal of constraint-based approaches toward understanding molecular interaction networks should be to develop a theory capable of expressing the manner in which information is stored and transmitted in networks of molecular interactions~\cite{Tkacik2011a,Bialek2012} and how such processes produce the higher-order properties such as phenotypes that serve as substrate for natural selection~\cite{Mora2013}. A canonical information-theoretic approach has already been applied empirically to biochemical signal transduction networks~\cite{Cheong2011,Brennan2012}. Subtle but essential developments in information theory and its applications makes connections between molecular networks and networks of neurons immediate~\cite{Stanley1999,Balduzzi2008,Balduzzi2009,Balduzzi2012}, while also supporting generalization from molecular networks to organisms and their populations in application to evolutionary processes~\cite{Kussell2005a,Rivoire2011}. Common to all of these instantiations is the drive toward understanding the manner in which relationships between some interacting entities and the mutual constraints implicit to such interactions can be represented mathematically in both algebraic and geometric form. While this study has provided another instantiation of such an approach, it seems to be a mere fragment of a deeper theory requiring expression in terms of a language that curious specialists from each of the relevant fields can all understand well enough to speak.
