The genotype of an organism has a relatively straightforward definition in terms of the sequence of nucleotides comprising its genome. Phenotypes, on the other hand, can be described at different levels of organization~\cite{Stadler2001}. For example, one might consider the lowest level description of phenotype as a dynamic phenomenon requiring specification of the time-dependent transcription rates of all genes comprising an organism's genome. Historically, this level of resolution has been impossible to achieve empirically and the phenotype concept was initially defined at the level of macroscopically observable physical characteristics such as shape, size, color, and various combinations thereof. The present capability to assess phenotypes at various levels of organization raises several fundamental questions about genotype-phenotype mappings that can be described inductively. One is to define and characterize properties of the fundamental level of genotype-phenotype mapping that is embodied in the transcription process. The next is to define and characterize the manner in which higher-order "phenotype-phenotype" mappings built on top of this one are affected by the properties of each of the levels existing below, and, on evolutionary timescales over which feedback may play a significant role, potentially also those above it.

These questions can be addressed to some extent from an abstract perspective that is independent of the particular physical implementation of various genotype-phenotype mappings in terms of complex networks of molecular interactions. Here we describe one formalization of the genotype-phenotype mapping that takes into account evidence that at the transcriptional, and thus potentially most fundamental, level the genotype-phenotype mapping is stochastic \cite{Swain2002,Paulsson2004,Thattai2004,Acar2008a,Lestas2010,Munsky2012,Neuert2013,Sanchez2013}. The general formalization we describe allows us to precisely pose the following question: \emph{For any genotype, given a space of probability distributions on the collection of genotype-phenotype maps, does modularization over that genotype provide access to a lesser, equivalent or greater collection of correlations among the expression patterns of genes that define phenotypes?} Modularization here refers to restricting interactions among genes in a particular way. For example, if we have four genes which can all participate in one potentially repeated higher-order (e.g. ternary) interaction that generates a phenotype, but we restrict interactions to a subset consisting of four of the six possible binary interactions then this represents one particular way of modularizing over a given genotype.

If modularization were to ultimately result in a space of probability distributions over genotype-phenotype maps that could be included in or placed into one-to-one correspondence with the space of probability distributions over the genotype-phenotype mappings defined on the full genotype (i.e. non-modular), then one would not expect there to necessarily be a competitive advantage for organisms incorporating molecular mechanisms that explicitly impose such modularity. The relative abundance of organisms that do precisely this would then appear to be paradoxical \cite{Jothi2009,Colm}. If, on the other hand, modularization provides access to a larger space of probability distributions over genotype-phenotype mappings than for the non-modular case, then one can imagine that certain environmental conditions would be unable to be addressed by the functions capable of being achieved by the gene expression profiles that can be achieved by non-modular genomes.

We provide a precise quantitative answer to the question stated above for one particular kind of modularization over a genotype. The answer we suggest has the qualitative implication that modularization over a genotype may provide access to a \emph{larger} collection of potential gene expression patterns and thus a larger collection of potential biological functions that may derive from this expanded repertoire of patterns.
%In doing so, we make use of a fragment of the mathematical language of category theory. Expressing our results in this language facilitates expanding the domain of applicability of the model described. In addition to the direct value of using category theory as a framework for organizing information for the purpose of constructing precise modeling descriptions, this approach provides an example of sharing modeling constructions between different fields of science.
