%!TEX root = ../plos_template.tex
The genotype of an organism has a relatively straightforward definition in terms of the sequence of nucleotides comprising its genome. Phenotypes, on the other hand, can be described at different levels of organization~\cite{Dawkins1982,Stadler2001}. The concept of phenotype was initially defined at the level of macroscopically observable physical characteristics~\cite{Johannsen1911}. The lowest level description of phenotype might be considered to be the dynamic phenomenon requiring specification of time-dependent transcription rates of all genes comprising an organism's genome. The present capability to assess phenotypes at various levels of organization raises several questions about genotype-phenotype mappings. One is how to define and characterize properties of the fundamental level of genotype-phenotype mapping that is embodied in the transcription process. Another is how to define and characterize the manner in which higher-order ``phenotype$_i$-phenotype$_{i+1}$'' mappings built on top of this one are affected by the properties of each of the levels existing below, and, on evolutionary timescales over which feedback may play a significant role, potentially also those above it.

We present a somewhat formal description of the genotype-phenotype mapping that takes into account evidence at the transcriptional level that the genotype-phenotype mapping is stochastic \cite{Swain2002,Paulsson2004,Thattai2004,Acar2008a,Lestas2010,So2011,Munsky2012,Neuert2013,Sanchez2013} in order to begin to assess the impact of this feature on the evolution of hierarchical modularity in gene regulatory networks. The general formalism we describe allows us to precisely pose the following question: \emph{For any collection of genes, given a space of probability distributions on the associated collection of genotype-phenotype maps, does modularization over that collection of genes increase, decrease, or leave invariant the size of the space of gene expression patterns that define low-level phenotypes and can thus be selected via external constraints?} Modularization here refers to restricting interactions among genes in a particular manner encoded by a hypergraph (Supplementary Material).

If modularization were to ultimately result in no change to the space of probability distributions over genotype-phenotype maps, then there would be no way to distinguish the collection of possible gene expression patterns of organisms incorporating molecular mechanisms that explicitly impose such modularity, and thus no means of selecting for modular or non-modular genomes~\cite{Jothi2009,Colm}. If, on the other hand, modularization provides access to a larger or smaller space of probability distributions over genotype-phenotype mappings, then certain environmental conditions would be capable of selectively discriminating among certain classes of network architecture. We provide a precise answer to the question stated above for all possible modularizations of interaction over a collection of genes.

% In a network context in which all interaction orders are allowed, restriction of higher-order interactions within the underlying process intuitively reduces the size of the space of accessible gene expression patterns. However, in network contexts that are constrained to match the stationary structure of the underlying process, modularizations over a genotype that include at least one cycle in the hypergraph representing the resulting modular structure provide access to a \emph{larger} collection of potential gene expression patterns and thus a larger collection of potential biological functions that may derive from the expanded repertoire of such patterns. We provide a precise quantitative answer to the question stated above for several different modularizations over a collection of interacting genes.
