%!TEX root = ../plos_template.tex
% The genotype of an organism has a relatively straightforward definition in terms of the sequence of nucleotides comprising its genome. Phenotypes, on the other hand, can be described at different levels of organization~\cite{Dawkins1982,Stadler2001}. The concept of phenotype was initially defined at the level of macroscopically observable physical characteristics such as shape, size, color, and various combinations thereof~\cite{Johannsen1911}. The lowest level description of phenotype might be considered to be the dynamic phenomenon requiring specification of, potentially time-dependent, transcription rates of all genes comprising an organism's genome. The present capability to assess phenotypes at various levels of organization raises several questions about genotype-phenotype mappings. One is how to define and characterize properties of the fundamental level of genotype-phenotype mapping that is embodied in the transcription process. Another is how to define and characterize the manner in which higher-order ``phenotype$_i$-phenotype$_{i+1}$'' mappings built on top of this one are affected by the properties of each of the levels existing below, and, on evolutionary timescales over which feedback may play a significant role, potentially also those above it.

Traditional dynamical models of gene-regulatory networks begin by treating them as reaction networks whose known or hypothesized interactions are encoded by sigmoidal response functions \cite{Alon2006,Voit2012}. Genes that interact do so in such a manner as to either activate or inhibit one another leading to a general deterministic dynamical model.
% \begin{eqnarray*}
% {dg_i \over dt} =
% k_{is}  \prod_{j=1}^{N} \frac{k_{ija} g_j^{n_{ija}}}{1+k_{ija} g_j^{n_{ija}}}
% \frac{1}{1+k_{ijr} g_j^{n_{ijr}}} - k_{id} g_i
% \end{eqnarray*}
% for a network of $N$ genes where $g_i$s represent concentrations of gene products, $k_{ija}$s represent activation rates, $k_{ijr}$s represent repression rates, and $k_{id}$s represent degradation rates.
The dual perspective is taken by statistical models whose parameters are fit to data obtained by measuring gene expression using microarray or sequence census methods \cite{Friedman2008a,Zhang2013}. In both of these classes of models, higher-order correlations are assumed to be accurately represented in terms of a sum over lower-order correlations.

% We attempt to address these questions from an abstract perspective that is independent of the particular physical implementation of genotype-phenotype mappings in terms of complex networks of molecular interactions.
We present a formal description of the genotype-phenotype mapping that in addition to taking into account evidence that, at the transcriptional level, this mapping is stochastic \cite{Swain2002,Paulsson2004,Thattai2004,Acar2008a,Lestas2010,So2011,Munsky2012,Neuert2013,Sanchez2013} also allows for the inclusion of higher-order correlations. In order to assess the impact of this feature on the evolution of gene regulatory network architecture, we use this description to precisely pose the question: For any genotype, given a space of probability distributions on the collection of genotype-phenotype maps, does modularization over that genotype increase, decrease, or leave invariant the size of the accessible space of gene expression patterns? Modularization here refers to restricting interactions among genes (see Supplementary Information).
% For example, if we have three genes that can all participate in a higher-order interaction (the highest-order among three genes being a trinary interaction) that generates a phenotype, but we restrict interactions to a lower-order subset consisting of all three possible binary interactions then this represents one particular way of modularizing over a given genotype (compare \ref{fig:conediagram}B top and middle).

For those cases in which modularization results in no change to the space of probability distributions over genotype-phenotype maps with respect to that defined on the full genotype (i.e. the single highest-order interaction referred to as non-modular), then there may be no method to select for one network architecture over another. On the other hand any differences in gene expression patterns that result from differences in network architecture may allow certain network architectures to be selected relative to one another. Here we show that those modularizations that introduce cycles into the gene-regulatory network architecture also expose the capacity for negative selection as a result of additional restrictions upon the gene expression patterns that apply to networks with cycles but not to those without.

% then there would be no way to observationally distinguish the collection of possible gene expression patterns of organisms incorporating molecular mechanisms that explicitly impose such modularity, and thus no means of selecting specifically for modular or non-modular gene regulatory networks~\cite{Jothi2009,Colm}. If, on the other hand, modularization provides access to a larger or smaller space of probability distributions over genotype-phenotype mappings than for the non-modular case, then one can imagine that certain environmental conditions would be unable to be addressed by the functions capable of being achieved by the gene expression profiles accessible to one gene-regulatory network with respect to another. We provide a precise quantitative answer to the question stated above for several different modularizations over a collection of interacting genes.

% In a network context in which all interaction orders are allowed, restriction of higher-order interactions within the underlying process intuitively reduces the size of the space of accessible gene expression patterns. However, in network contexts that are constrained to match the stationary structure of the underlying process, modularizations over a genotype that include at least one cycle in the hypergraph representing the resulting modular structure provide access to a \emph{larger} collection of potential gene expression patterns and thus a larger collection of potential biological functions that may derive from the expanded repertoire of such patterns. We provide a precise quantitative answer to the question stated above for several different modularizations over a collection of interacting genes.
