%!TEX root = ../plos_template.tex
The genotype of an organism has a relatively straightforward definition in terms of the sequence of nucleotides comprising its genome. Phenotypes, on the other hand, can be described at different levels of organization~\cite{Dawkins1982,Stadler2001}. For example, one might consider the lowest level description of phenotype as a dynamic phenomenon requiring specification of, potentially time-dependent, transcription rates of all genes comprising an organism's genome. The concept of phenotype was initially defined at the level of macroscopically observable physical characteristics such as shape, size, color, and various combinations thereof. The present capability to assess phenotypes at various levels of organization raises several questions about genotype-phenotype mappings. One is how to define and characterize properties of the fundamental level of genotype-phenotype mapping that is embodied in the transcription process. Another is how to define and characterize the manner in which higher-order ``phenotype$_i$-phenotype$_{i+1}$'' mappings built on top of this one are affected by the properties of each of the levels existing below, and, on evolutionary timescales over which feedback may play a significant role, potentially also those above it.

We attempt to address these questions from an abstract perspective that is independent of the particular physical implementation of genotype-phenotype mappings in terms of complex networks of molecular interactions. Here we present a somewhat formal description of the genotype-phenotype mapping that takes into account evidence that at the transcriptional, and thus potentially most fundamental, level the genotype-phenotype mapping is stochastic \cite{Swain2002,Paulsson2004,Thattai2004,Acar2008a,Lestas2010,So2011,Munsky2012,Neuert2013,Sanchez2013} in order to begin to assess the impact of this feature on the evolution of hierarchical modularity in gene regulatory networks. The general formalization we describe allows us to precisely pose the following question: \emph{For any genotype, given a space of probability distributions on the collection of genotype-phenotype maps, does modularization over that genotype increase, decrease, or leave invariant the size of the accessible space of gene expression patterns that define low-level phenotypes?} Modularization here refers to restricting interactions among genes in a particular way (see Supplementary Information). For example, if we have three genes that can all participate in a higher-order interaction (the highest-order among three genes being a trinary interaction) that generates a phenotype, but we restrict interactions to a lower-order subset consisting of all three possible binary interactions then this represents one particular way of modularizing over a given genotype (compare \ref{fig:conediagram}B top and middle).

If modularization were to ultimately result in no change to the space of probability distributions over genotype-phenotype maps with respect to that defined on the full genotype (i.e. the single highest-order interaction referred to as non-modular), then there would be no way to distinguish the collection of possible gene expression patterns of organisms incorporating molecular mechanisms that explicitly impose such modularity, and thus no means of selecting for modular or non-modular genomes. The relative abundance of organisms that exhibit modular structure would then appear to be paradoxical \cite{Jothi2009,Colm}. If, on the other hand, modularization provides access to a larger space of probability distributions over genotype-phenotype mappings than for the non-modular case, then one can imagine that certain environmental conditions would be unable to be addressed by the functions capable of being achieved by the gene expression profiles accessible to non-modular gene-regulatory network architectures.

We provide a precise quantitative answer to the question stated above for several different modularizations over a collection of interacting genes (specifically those of \ref{fig:ncycvolrat}). The answer we suggest has the qualitative implication that modularizations over a genotype that include at least one cycle in the underlying hypergraph representing the resulting modular structure may provide access to a \emph{larger} collection of potential gene expression patterns and thus a larger collection of potential biological functions that may derive from the expanded repertoire of such patterns.
%In doing so, we make use of a fragment of the mathematical language of category theory. Expressing our results in this language facilitates expanding the domain of applicability of the model described. In addition to the direct value of using category theory as a framework for organizing information for the purpose of constructing precise modeling descriptions, this approach provides an example of sharing modeling constructions between different fields of science.
