%!TEX root = ../plos_template.tex
\section{Cyclic network contexts can impose unsatisfiable constraints}\label{sec:cycliccontextunsatisfiableconstraints}

% We can use an undirected graph (e.g. \ref{fig:inconsistentthreecycle}A, bottom left) to indicate the manner in which a given collection of genes is connected to the network context (\ref{fig:stochdynscheme}, bottom row). We refer to this as the GRNA (see \ref{sec:covergenotypespace}).

Each node of the \SH{} graph can be associated to the probability distribution that specifies probabilities for each biological variable to be observed in each of the states determined by the coarse-graining process described in \ref{sec:genenetworkphenmap}. Each edge of the graph specifies a joint probability distribution for both of the nodes it contains (or connects) to simultaneously take on a given pair of values. Note that this does not imply the existence or absence of a physical interaction between the genes represented by these two nodes. Together, these probabilities represent constraints that the network context may impose upon the network.
%We generalize this in what proceeds to hypergraphs where each edge can contain an arbitrary number of nodes because this is necessary in order to take into account all possible interactions among genes. In this context it is presumed that via marginalization the probability distributions associated to each edge are consistent with the probability distributions associated to each node. This is similar to using a Markov Random Field (a particular type of probabilistic graphical model) as a model for the stationary distribution of the stochastic process underlying gene expression \cite{Lauritzen1996,Koller2009,Chen2013a}.
%However, the implication of existence of a factorizable joint probability distribution over all nodes in the graph, which is implicit in the machine learning interpretation of graphical models, is not assumed here (\refsupp{}) \cite{Barber2012,Bishop2007,Murphy2012,Koller2009}.
\ref{fig:inconsistentthreecycle}A provides a concrete example where we assume three genes are observed via all possible pairwise combinations and that via the coarse-graining process we have binned the expression state of each gene into one of two classes.
%Note that this graphical model is completely general in the sense that we can have an arbitrary number of genes and an arbitrary number of classes in the assignment of expression states up to the maximal resolution of the method of observation.
Each node of the graph in \ref{fig:inconsistentthreecycle}A represents a probability distribution over the observation of each gene in either of the two states established in the coarse-graining process.
%, which for node three are given by $p((3 \mapsto 0,\,\, 3 \mapsto 1))=(0.5,0.5)$. Given an edge we consider a map or function that takes as input the two genes connected by or contained within that edge and maps those two genes simultaneously to the available expression states. For example in \ref{fig:inconsistentthreecycle}A bottom there are maps $12 \mapsto 11$, $23 \mapsto 10$, and $31 \mapsto 10$ with associated probabilities $p(12 \mapsto 11)=0.1$ \ref{fig:inconsistentthreecycle}A top purple box and bottom purple arrows, $p(23 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top orange box and bottom orange arrows, and $p(31 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top green box and bottom green arrows.
Each of the probability tables adjacent to each edge in the graph assigns a probability distribution to the set of maps from the nodes connected by the edge to all possible combinations of the expression states. As these maps take collections of genes as input and produce collections of expression states as outputs we refer to them as \gnpm{} and thus to the associated probability distributions as probability distributions over \gnpm{}.

Suppose the normalized contingency tables in \ref{fig:inconsistentthreecycle}A are meant to represent the ostensible structure and parameters of a gene expression process. It is often necessary to attempt to infer the parameters of such a model from data under the assumption that the structure of a given GRN falls within the model class defined by a given graph. \ref{fig:inconsistentthreecycle}B represents a case in which a hypothetical dataset is consistent with its derivation from a joint probability distribution whereas \ref{fig:inconsistentthreecycle}C represents a case of inconsistency where the pairwise distributions are each individually consistent distributions, but, together, the three pairwise distributions are not consistent with any joint distribution over the states of all three genes.
%\ref{fig:inconsistentthreecycle}B top left represents hypothetical data containing three hundred observations generated by a process whose structure is equivalent to that of \ref{fig:inconsistentthreecycle}A except that we consider the case in which each expression state is equally likely. We now use maximum likelihood estimation \cite{Barber2012} to detertmine the parameters of a model like that of \ref{fig:inconsistentthreecycle}A given the dataset in \ref{fig:inconsistentthreecycle}B top left. \ref{fig:inconsistentthreecycle}B top middle shows that indeed all possible binary sequences representing the expression states for genes one, two and three are approximately uniformly represented in the given samples and the green bars in \ref{fig:inconsistentthreecycle}B bottom show the marginals of this joint distribution. Alternatively, we can infer the marginal probabilities using the sum-product (belief propagation) algorithm \cite{Barber2012} resulting in the yellow bars in \ref{fig:inconsistentthreecycle}B bottom. If we perform the same sequence of steps on the hypothetical data in \ref{fig:inconsistentthreecycle}C top left, only the (loopy) belief propagation algorithm returns marginal distributions (\ref{fig:inconsistentthreecycle}C bottom) that are consistent with the data. This is a result of the fact that no joint probability distribution having these empirical marginals exists and maximum likelihood estimation determines a joint distribution that only approximates the empirical marginals. Dually, only the data of \ref{fig:expression_concept}C bottom and not that of \ref{fig:expression_concept}C top is consistent with the probabilities represented in \ref{fig:inconsistentthreecycle}A.
This inconsistency is made possible by the fact that the network architecture in \ref{fig:inconsistentthreecycle}A contains a cycle \cite{Lauritzen1996,Geiger2006,Wainwright2007} and that we have given an ostensible data set leading to the inference of parameters that could not possibly derive from a joint probability distribution over all three genes.

If this situation arises, it indicates some systematic error in the transfer of information whether it occurs intrinsically to the system wherein a network has inconsistent constraints placed upon it by its network context or as part of the scientific data collection process. In the former case, this can be resolved by modifying the inconsistent constraints in such a manner that they become consistent with or without modifying the network architecture in doing so. In the latter case, this may result from employing a model which 1) takes insufficient account of the network context and 2) relies on coarse-grained observations. In either case, the synthetic gene circuit schematized in \ref{fig:condgenescenario} serves as one mechanism implementing the example presented in \refsupp{} \ref{secsupp:apparentinconsistency}. It consists of four genes each of which is capable of taking on three different states \cite{Rieckh2013a}. However, observing two out of the three states measured pairwise from three out of the four genes could result in data that would appear to be inconsistent. Such an observation would demonstrate without having to have knowledge of the correct network architecture, that the current model is insufficient to represent the underlying process.

For the case of the architecture in \ref{fig:inconsistentthreecycle}A, and moreover for any GRNA of any size that contains one or more cycles, the possibility of finding a joint distribution over all genes that satisfies all constraints capable of being imposed upon it requires the implicit assumption that the structure of the network context can be viewed simultaneously as that of \ref{fig:expression_concept}C top and that of \ref{fig:expression_concept}C bottom. The spaces of probability distributions corresponding to the constraints that can be imposed upon the two GRNAs contrasted in \ref{fig:expression_concept}C are different. We can now apply the process described in \ref{sec:compatibilityofgpms} to classify the geometries and thus relationships among the spaces of probability distributions associated to constraints that can be imposed on all possible GRNAs with a given number of genes.

\section{Geometry of probabilistic constraints on network states}\label{sec:probconstrgeometry}
The collection of all possible GRNAs is equivalent to the lattice of all possible reduced subsets of genes (\ref{sec:covergenotypespace} and \refsupp{}). For example, \ref{fig:conediagram}A contains the lattice of reduced subsets of three genes. We are only interested in those subsets that contain at least one copy of each gene, which corresponds to the region highlighted with a gray background in \ref{fig:conediagram}A. Each GRNA corresponds to a different modularization of the \gnpm{} by the network context. For example, \ref{fig:conediagram}B shows in the same vertical order the different maps induced by the three architectures highlighted in green in \ref{fig:conediagram}A.

We consider those GRNAs found lower in the lattice of \ref{fig:conediagram}A to be of higher modularity because each corresponds to the increasing restriction from placing constraints on higher- to placing constraints on lower-order correlations among genes. \ref{fig:conediagram}B top corresponds to the least modular GRNA because constraints are placed upon correlations among all three genes. \ref{fig:conediagram}B middle exhibits an elevated degree of modularity because constraints are placed upon correlations among pairs of genes. Similarly, \ref{fig:conediagram}B bottom is even more modular because constraints are placed upon each gene individually.

Each of the GRNAs in \ref{fig:conediagram}A can be associated to a space of probability distributions over \gnpm{}. \ref{fig:conediagram}C schematically depicts the relationships among the probability distributions associated to the corresponding architectures and \gnpm{} in \ref{fig:conediagram}B. The inconsistency noted in the previous section between the architectures \ref{fig:conediagram}B top and middle is a result of the differing geometries in \ref{fig:conediagram}C middle. There, the smaller darker gray region defined by the inequalities expressed in \ref{eq:threecycinequalities} and \ref{eq:threecycbooleinequalities} corresponds to the space of probability distributions defined over all possible \gnpm{} associated to the network architecture in \ref{fig:conediagram}B top. Similarly, the lighter gray region defined by \ref{eq:threecycinequalities} alone corresponds to the analogous structure for \ref{fig:conediagram}B middle.

% This relationship indicates that there are certain cases in which a given GRNA exhibiting a relatively higher degree of modularity nevertheless provides access to a larger space of possible probability distributions and thus of correlations among genes which can be coarsely referred to as gene expression patterns. These patterns may serve as substrate for natural selection of GRNAs with particular functions by the network context within which the small modules we have considered so far function.

\section{Naive likelihood of sampling unsatisfiable constraints}\label{sec:volrat}
Relationships between spaces of potential constraints placed upon gene expression patterns like that of \ref{fig:conediagram}C middle occur for all GRNAs on any number of genes so long as there exists at least one cycle in the corresponding network architecture. For the case of three genes, there is only one class of graphs containing a cycle, which is that of \ref{fig:conediagram}B middle. For the case of four genes there are nine different classes of hypergraphs containing cycles and these nine classes can be split into two groups depending upon whether or not the edges of the graphs are each restricted to represent correlations among only two genes. \ref{fig:non2uniformcyclichypergraphhasse} shows the components of the lattice analogous to that of \ref{fig:conediagram}A.
% containing the five out of the nine classes of hypergraphs on four genes that have cycles but that are not 2-uniform (the hypergraphs where at least one edge is used to represent interactions among more than two genes).

Given this larger collection of GRNAs with cycles we can assess the relative sizes of the analogs to the light and dark gray spaces (\ref{fig:conediagram}C middle) of probability distributions over \gnpm{}. We assess the likelihood of choosing a point in the dark gray region at random by computing the ratio of the volume of the dark gray region (associated to the non-modular GRNA analogous to that of \ref{fig:conediagram}B top with a single edge containing all four genes), whose architecture and thus volume is fixed, to that of the light gray region, whose volume varies according to each of the cyclic graphs associated to a GRNA on four genes. We refer to this number as the global:local volume ratio or~$\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}$~(see \ref{sec:compatibilityofgpms} and \refsupp{}). The comparison defined by this ratio is meaningful since $\mathbb{L}(\mathcal{G})$, \ref{eq:localpolytope}, and $\mathbb{M}(\mathcal{G})$, \ref{eq:globalpolytope} are of the same dimension. In the case where the constraints defining $\mathbb{L}(\mathcal{G})$ are eliminated, the analog of this volume ratio would be $0$ for all $\mathcal{G}$. This volume ratio determines the \emph{a priori} likelihood of observing inconsistency for a given GRNA. The consistency check involved in computing this ratio can be used as a test demonstrating, for those cases exhibiting inconsistency, that the model being used is incorrect in the sense that it does not correspond sufficiently to the actual network context determining the constraints placed upon the network. Consider the probability of locally versus globally consistent \emph{observations} ($p(\mathbb{L}(\mathcal{G})_O)$  vs $p(\mathbb{M}(\mathcal{G})_O)$ respectively) separately from the probability of locally versus globally consistent \emph{models} ($p(\mathbb{L}(\mathcal{G})_M)$  vs $p(\mathbb{M}(\mathcal{G})_M)$ respectively) that accurately reflect the underlying process. We can then estimate the probability of having a locally consistent model despite obtaining globally consistent observations, $p(\mathbb{L}(\mathcal{G})_M | \mathbb{M}(\mathcal{G})_O)$, via a simple application of Bayes' theorem
$$
p(\mathbb{L}(\mathcal{G})_M | \mathbb{M}(\mathcal{G})_O) = \frac{p(\mathbb{M}(\mathcal{G})_O | \mathbb{L}(\mathcal{G})_M)p(\mathbb{L}(\mathcal{G})_M)}{p(\mathbb{M}(\mathcal{G})_O | \mathbb{L}(\mathcal{G})_M)p(\mathbb{L}(\mathcal{G})_M) + p(\mathbb{M}(\mathcal{G})_O | \mathbb{M}(\mathcal{G})_M)p(\mathbb{M}(\mathcal{G})_M)},
$$
where $p(\mathbb{M}(\mathcal{G})_O | \mathbb{M}(\mathcal{G})_M)=1$, the volume ratio described above corresponds to $p(\mathbb{M}(\mathcal{G})_O | \mathbb{L}(\mathcal{G})_M)$, and one could consider the impact of different prior probabilities, $p(\mathbb{L}(\mathcal{G})_M)$, of having a locally consistent model.

\ref{fig:ncycvolrat}A shows the results of computations for seven different graphs where the edges are restricted to each contain two genes. The spaces are equivalent and thus the volume ratio equal to one for graphs lacking cycles (e.g. the first three graphs along the $x$-axis of \ref{fig:ncycvolrat}A). For the four network architectures in \ref{fig:ncycvolrat}A containing cycles, the volume ratio is strictly less than one. This quantifies the probability that the network architecture depicted along the $x$-axis will be able to satisfy the constraints that the associated network context is capable of placing upon it.

\section{Potential for unsatisfiable constraints may bias the sampling of network architectures by evolutionary processes}\label{sec:unsatisfiableconstrevolution}

% The difference in the satisfiability of constraints capable of being placed on the various architectures with cyclic components, as compared to the patterns available to the network architecture possessing correlations among all genes, is equivalent to the statement that certain probability distributions over \gnpm{} are inaccessible to certain classes of GRNAs. This indicates that if the context (see \ref{fig:stochdynscheme} left) within which the network modules we have considered so far are embedded places certain functional requirements upon them, then the ability of any GRNA to be able to meet those requirements must be considered relative to the geometry of its associated space of probability distributions.

The satisfiability of constraints capable of being placed on the various architectures is logically a function of whether or not the GRNA is cyclic or acyclic. For those GRNAs containing cycles, there are certain functional requirements that can be achieved so long as only local and not global consistency is required of them. Once global consistency is imposed as in the structure corresponding to the joint correlations among all genes, those functions that were accessible when only local consistency was imposed are unavailable. For acyclic GRNAs, there is no difference between the satisfiability of locally or globally imposed constraints. \ref{fig:stochdynscheme} right shows a schematic of one potential scenario by which a given cyclic GRNA may be selected against. The black points in the center represent an initial condition of a stochastic process that is selected for its ability to achieve one of two different stationary distributions represented by the blue and the red points respectively. This is equivalent to placing a fitness landscape given by a function whose maximum is located at the given points and defined over the relevant space of probability distributions. The GRNA represented in the top row of \ref{fig:stochdynscheme} is able to achieve as its stationary distribution any of the constraints capable of being imposed upon it that are consistent with its architecture because it is acyclic. On the other hand, the GRNA in the bottom row is incapable of achieving certain constraints that may be imposed upon it by a network context consistent with its architecture because it is cyclic.

When selective pressure is induced equivalent to the distribution located at the blue point, or at any other point within the dark gray region, either of the architectures are essentially equivalent with respect to the statistics of samples from their corresponding probability distributions and they can thus be considered as members of an evolutionarily neutral space. On the other hand, selective pressure equivalent to the probability distributions located at the red point differentiates between the networks of the top and bottom row or equivalently between the network of the bottom row when global consistency is imposed versus the same network when only local consistency conditions are imposed. The same qualitative relationship holds true for the spaces of probability distributions of all GRNAs of any size and for any number of different expression levels so long as the graph associated to the relevant correlations among genes contains at least one cycle.

The distinction between cyclic and acyclic GRNAs with respect to the ability to have unsatisfiable constraints placed upon them is sharp. However, within the class of cyclic GRNAs, the likelihood of having unsatisfiable constraints imposed on a given GRNA increases, at least approximately, with the number of cycles in the given GRNA (\ref{fig:ncycvolrat} and \ref{sec:volrat}). This indicates that the strength of selection against GRNAs with a larger number of nested cycles is likely to be stronger than that against GRNAs with a relatively smaller number of cycles. Initiating an evolutionary process with a large network containing many nested cycles may then result in the elimination of some via gene duplication until the number of nested cycles decreases sufficiently so that the intrinsic strength of selection against cycles that we have characterized here reaches equilibrium with the rate at which new cycles form. One possibility, depending upon the overall relationship between these rates, is a globally hierarchical network with a number of cyclic modules interspersed throughout.
