%!TEX root = ../plos_template.tex
\subsection{Coarse-graining gene expression}
In any attempt to model gene regulatory networks, a focal subnetwork is conceptually carved out of a larger network context. The output of the subnetwork, or some function of it, may serve as input to the network context and vice versa \ref{fig:stochdynscheme}. Each component of the network context may have access to different pieces of partial information from the focal subnetwork depending upon the manner in which each is connected. The imposition of functional requirements by the network context upon the subnetwork is equivalent both to imposing a fitness landscape on the outputs of the subnetwork and to requiring that certain patterns of output from the subnetwork be observable. The manner in which the subnetwork is connected to the network context determines the structure of observations whereas the manner in which connections are made among genes within the subnetwork determine the structure of the process that underlies the repertoire of functions the network is capable of displaying independent of its context. We describe a simple representation of observing gene expression in a manner capable of mimicking the embedding of a given subnetwork within any possible network context.

\ref{fig:expression_concept}A shows a simplified representation of a bacterial chromosome or plasmid encoding a small gene regulatory network (GRN) whose correlations strengths are not known but are to be derived from observation of the transcription process. The amount of a given transcript present in a cell given in terms of the discrete counts obtained via sequence census methods (e.g. RNA-seq) or relative abundance derived from microarray data can be binned into a smaller number of discrete classes by setting a collection of thresholds on the original data set. If only a single threshold is given, then the data can be binned into two classes depending upon whether or not the original measurement surpasses the given threshold \ref{fig:expression_concept}B. If a large enough number of thresholds is available to distinguish among all possible molecule counts, then this observational protocol becomes complementary to mechanistic models of stochastic gene expression that seek solutions of the master equation, which are probability distributions over molecule counts \cite{Walczak2009,Mugler2009}. For example, in the case of a three gene network, the master equation takes the form
$$
\frac{dP(n_1,n_2,n_3)}{dt} = M(k) P(n_1,n_2,n_3)
$$
where $P(n_1,n_2,n_3)$ gives the probability of observing $n_1$, $n_2$, and $n_3$ molecules of each of the three genes respectively and $M(k)$ is a Markov transition rate matrix that depends upon some rate functions $k$ that are determined by the network architecture and the dynamics of the interactions. $\frac{d}{dt}P_{s}(n_1,n_2,n_3)=0$ for the case of a stationary distribution $P_s$. We can then coarse-grain over the molecule counts using a function that maps the states represented by vectors of integers into some other variables. For example, if $n_i$ are positive integers, then a function $f \colon \mathbb{Z}^+ \rightarrow \{0,1\}$ that takes any integer less than or equal to some threshold $T \in \mathbb{Z}^+$ to $0$ and any integer greater than $T$ to $1$ is a very simple example of such a coarse-graining. In this case, that is for this specific form of the coarse-graining function $f$, the coarse-grained stationary probability distribution takes the form
$$
P_s(b_1,b_2,b_3) = \sum_{n_1 \in f^{-1}(b_1)}\sum_{n_2 \in f^{-1}(b_2)}\sum_{n_3 \in f^{-1}(b_3)} P_s(n_1,n_2,n_3).
$$

At the lowest-level of the genotype-phenotype mapping, these coarse-grained expression values can be viewed as collectively determining the lowest level in the aforementioned hierarchy of phenotypes. In this way, the data derived from a number of samples of a given network can be described using binary sequences \ref{fig:expression_concept}C.
%For a given number of genes, it is possible to sample any subset of genes and the structure of this sampling process plays a role in determining the interpretation of the structure of the underlying process that generates the data, which, in this case, includes all regulatory elements that impinge upon the GRN module under consideration. For example, if, whenever we sample, we always consider the expression state of three out of three genes, we imply that the underlying process can be represented as a joint probability distribution over those three genes \ref{fig:expression_concept}C top. If instead we consider each pair of genes separately \ref{fig:expression_concept}, we imply the existence of a different kind of structure, which is the sum or union of several simpler spaces of probability distributions, each representing lower order interactions within the network that has been conceptually carved out of a larger network context.

\subsection{Inconsistency between network architecture and context}
We can use an undirected graph (e.g. \ref{fig:inconsistentthreecycle}A, bottom left) to indicate correlations imposed by various GRN architectures (GRNAs). Each node of the graph is associated to the probability distribution that specifies probabilities for each gene to be observed in each of the states determined by the coarse-graining process described in the previous section. Each edge of the graph specifies a joint probability distribution for both of the nodes it contains (or connects) to simultaneously take on a given pair of values.
%We generalize this in what proceeds to hypergraphs where each edge can contain an arbitrary number of nodes because this is necessary in order to take into account all possible interactions among genes. In this context it is presumed that via marginalization the probability distributions associated to each edge are consistent with the probability distributions associated to each node. This is similar to using a Markov Random Field (a particular type of probabilistic graphical model) as a model for the stationary distribution of the stochastic process underlying gene expression \cite{Lauritzen1996,Koller2009,Chen2013a}.
%However, the implication of existence of a factorizable joint probability distribution over all nodes in the graph, which is implicit in the machine learning interpretation of graphical models, is not assumed here (Supporting Information) \cite{Barber2012,Bishop2007,Murphy2012,Koller2009}.
\ref{fig:inconsistentthreecycle}A provides a concrete example where we assume three genes are correlated via all possible pairwise combinations and that via the coarse-graining process we have binned the expression state of each gene into one of two classes.
%Note that this graphical model is completely general in the sense that we can have an arbitrary number of genes and an arbitrary number of classes in the assignment of expression states up to the maximal resolution of the method of observation.
Each node of the graph in \ref{fig:inconsistentthreecycle}A represents a probability distribution over the observation of each gene in either of the two states established in the coarse-graining process.
%, which for node three are given by $p((3 \mapsto 0,\,\, 3 \mapsto 1))=(0.5,0.5)$. Given an edge we consider a map or function that takes as input the two genes connected by or contained within that edge and maps those two genes simultaneously to the available expression states. For example in \ref{fig:inconsistentthreecycle}A bottom there are maps $12 \mapsto 11$, $23 \mapsto 10$, and $31 \mapsto 10$ with associated probabilities $p(12 \mapsto 11)=0.1$ \ref{fig:inconsistentthreecycle}A top purple box and bottom purple arrows, $p(23 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top orange box and bottom orange arrows, and $p(31 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top green box and bottom green arrows.
Each of the probability tables adjacent to each edge in the graph assigns a probability distribution to the set of maps from the nodes connected by the edge to all possible combinations of the expression states. As these maps take collections of genes as input and produce collections of expression states as outputs we refer to them as genotype-phenotype mappings and thus to the associated probability distributions as probability distributions over genotype-phenotype maps.

The probability distribution in \ref{fig:inconsistentthreecycle}A is meant to represent the actual structure and particular parameters of a gene expression process. It is likewise necessary to infer the parameters of such a model from data under the assumption that the structure of a given GRN falls within the model class defined by a given graph. \ref{fig:inconsistentthreecycle}B represents a case in which a hypothetical dataset is consistent with its derivation from a joint probability distribution whereas \ref{fig:inconsistentthreecycle}C represents a case of inconsistency among lower-order correlations.
%\ref{fig:inconsistentthreecycle}B top left represents hypothetical data containing three hundred observations generated by a process whose structure is equivalent to that of \ref{fig:inconsistentthreecycle}A except that we consider the case in which each expression state is equally likely. We now use maximum likelihood estimation \cite{Barber2012} to detertmine the parameters of a model like that of \ref{fig:inconsistentthreecycle}A given the dataset in \ref{fig:inconsistentthreecycle}B top left. \ref{fig:inconsistentthreecycle}B top middle shows that indeed all possible binary sequences representing the expression states for genes one, two and three are approximately uniformly represented in the given samples and the green bars in \ref{fig:inconsistentthreecycle}B bottom show the marginals of this joint distribution. Alternatively, we can infer the marginal probabilities using the sum-product (belief propagation) algorithm \cite{Barber2012} resulting in the yellow bars in \ref{fig:inconsistentthreecycle}B bottom. If we perform the same sequence of steps on the hypothetical data in \ref{fig:inconsistentthreecycle}C top left, only the (loopy) belief propagation algorithm returns marginal distributions (\ref{fig:inconsistentthreecycle}C bottom) that are consistent with the data. This is a result of the fact that no joint probability distribution having these empirical marginals exists and maximum likelihood estimation determines a joint distribution that only approximates the empirical marginals. Dually, only the data of \ref{fig:expression_concept}C bottom and not that of \ref{fig:expression_concept}C top is consistent with the probabilities represented in \ref{fig:inconsistentthreecycle}A.
This inconsistency is made possible by the fact that the network architecture in \ref{fig:inconsistentthreecycle}A contains a cycle \cite{Lauritzen1996,Geiger2006,Wainwright2007} and that we have given an ostensible data set leading to the inference of parameters that could not possibly derive from a joint probability distribution over all three genes. For the case of the architecture in \ref{fig:inconsistentthreecycle}A, and moreover for any GRNA of any size that contains one or more cycles, the possibility of finding a joint distribution over all genes requires the implicit assumption that the structure of the generating process can be viewed simultaneously as that of \ref{fig:expression_concept}C top and that of \ref{fig:expression_concept}C bottom. The spaces of probability distributions associated to the two GRNAs contrasted in \ref{fig:expression_concept}C are different. This situation leads to the question of whether or not it is possible to classify the geometries and thus relationships among the spaces of probability distributions associated to all possible GRNAs on a given number of genes.

\subsection{The geometry of GRNA function}
The collection of all possible GRNAs is equivalent to the lattice of all possible reduced subsets of genes (Supporting Information). For example, \ref{fig:conediagram}A contains the lattice of reduced subsets of three genes. We are only interested in those subsets that contain at least one copy of each gene, which corresponds to the region highlighted with a gray background in \ref{fig:conediagram}A. Each GRNA corresponds to a different modularization of the genotype-phenotype maps. For example, \ref{fig:conediagram}B shows in the same vertical order the different maps induced by the three architectures highlighted in green in \ref{fig:conediagram}A.

We consider those GRNAs found lower in the lattice of \ref{fig:conediagram}A to be of higher modularity because each corresponds to the increasing restriction from higher- to lower-order correlations among genes. \ref{fig:conediagram}B top corresponds to the least modular GRNA because phenotype values are determined by correlations among all three genes. \ref{fig:conediagram}B middle exhibits an elevated degree of modularity because phenotypes are determined by correlations among pairs of genes. Similarly, \ref{fig:conediagram}B bottom is even more modular because phenotype values are determined by each gene individually.

Each of the GRNAs in \ref{fig:conediagram}A can be associated to a space of probability distributions over genotype-phenotype mappings. \ref{fig:conediagram}C schematically depicts the relationships among the probability distributions associated to the corresponding architectures and genotype-phenotype mappings in \ref{fig:conediagram}B. The inconsistency noted in the previous section between the architectures \ref{fig:conediagram}B top and middle is a result of the differing geometries in \ref{fig:conediagram}C middle. There, the smaller darker gray region corresponds to the space of all possible probability distributions for all possible parameters achievable via application of the linear transformation corresponding to marginalization to the probability distribution defined over all possible genotype-phenotype mappings associated to the network architecture in \ref{fig:conediagram}B top. Similarly, the lighter gray region corresponds to the analogous structure for \ref{fig:conediagram}B middle.

% This relationship indicates that there are certain cases in which a given GRNA exhibiting a relatively higher degree of modularity nevertheless provides access to a larger space of possible probability distributions and thus of correlations among genes which can be coarsely referred to as gene expression patterns. These patterns may serve as substrate for natural selection of GRNAs with particular functions by the network context within which the small modules we have considered so far function.

\subsection{Differing geometries indicate different functions}
Relationships between spaces of potential gene expression patterns like that of \ref{fig:conediagram}C middle occur for all GRNAs on any number of genes so long as there exists at least one cycle in the corresponding network architecture. For the case of three genes, there is only one class of graphs containing a cycle, which is that of \ref{fig:conediagram}B middle. For the case of four genes there are nine different classes of graphs containing cycles and these nine classes can be split into two groups depending upon whether or not the edges of the graphs are each restricted to represent correlations among only two genes. \ref{fig:non2uniformcyclichypergraphhasse} shows the components of the lattice analogous to that of \ref{fig:conediagram}A.
% containing the five out of the nine classes of hypergraphs on four genes that have cycles but that are not 2-uniform (the hypergraphs where at least one edge is used to represent interactions among more than two genes).

Given this larger collection of GRNAs with cycles we can assess the relative sizes of the analogs to the light and dark gray spaces (\ref{fig:conediagram}C middle) of probability distributions over genotype-phenotype maps. We assess the likelihood of choosing a point in the dark gray region at random by computing the ratio of the volume of the dark gray region (associated to the non-modular GRNA analogous to that of \ref{fig:conediagram}B top with a single edge containing all four genes), whose architecture and thus volume is fixed, to that of the light gray region, whose volume varies according to each of the cyclic graphs associated to a GRNA on four genes. We refer to this number as the non-modular:modular volume ratio or~$\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}$~(Supporting Information). \ref{fig:ncycvolrat}A shows the results of computations for seven different graphs where the edges are restricted to each contain two genes. The spaces are equivalent and thus the volume ratio equal to one for graphs lacking cycles (e.g. the first three graphs along the $x$-axis of \ref{fig:ncycvolrat}A). For the four network architectures in \ref{fig:ncycvolrat}A containing cycles, the volume ratio is strictly less than one. This quantifies the degree to which the space of probability distributions associated to the network architecture depicted along the $x$-axis contains regions representing constraints that the associated network architecture is capable of satisfying.

\subsection{Network context imposes constraints on network architecture}

The difference in the repertoires of gene expression patterns available to the various architectures with cyclic components, as compared to the patterns available to the network architecture possessing correlations among all genes, is equivalent to the statement that certain probability distributions over genotype-phenotype maps are inaccessible to certain classes of GRNAs. This indicates that if the context (see \ref{fig:stochdynscheme} left) within which the network modules we have considered so far are embedded places certain functional requirements upon them, then the ability of any GRN to be able to meet those requirements must be considered relative to the geometry of its associated space of probability distributions.

For those GRNAs containing cycles, there are certain functional requirements that can be achieved so long as only local and not global consistency is required of them (Supporting Information). Once global consistency is imposed as in the structure corresponding to the joint correlations among all genes, those functions that were accessible when only local consistency was imposed are unavailable. \ref{fig:stochdynscheme} right shows a schematic of one potential scenario by which a given GRNA may be selected against. The black points in the center represent an initial condition of a stochastic process that is selected for its ability to achieve one of two different stationary distributions represented by the blue and the red points respectively. This is equivalent to placing a fitness landscape given by a function whose maximum is located at the given points and defined over the relevant space of probability distributions. The GRNA represented in the top row of \ref{fig:stochdynscheme} is able to achieve as its stationary distribution any of the constraints capable of being imposed upon it that are consistent with its architecture. On the other hand, the GRNA in the bottom row is incapable of achieving certain constraints that may be imposed upon it by a network context consistent with its architecture.

When selective pressure is induced equivalent to the distribution located at the blue dot, or at any other point within the dark gray region, either of the architectures are essentially equivalent with respect to the statistics of samples from their corresponding probability distributions and they can thus be considered as members of an evolutionarily neutral space. On the other hand, selective pressure equivalent to the probability distributions located at the red point differentiates between the networks of the top and bottom row or equivalently between the network of the bottom row when global consistency is imposed versus the same network when only local consistency conditions are imposed. The same qualitative relationship holds true for the spaces of probability distributions of all GRNAs of any size and for any number of different expression levels so long as the graph associated to the relevant correlations among genes contains at least one cycle.
