%!TEX root = ../plos_template.tex
\subsection{Coarse-graining gene expression admits a simple probabilistic model}
In any attempt to model gene regulatory networks, a focal subnetwork is conceptually carved out of a larger network context. The output of the subnetwork serves as input to the network context. The impostion of functional requirements by the network context upon the subnetwork is equivalent to requiring that certain patterns of output of the subnetwork be observable. The manner in which the subnetwork is connected to the network context determines the interactions that specify the structure of observations whereas the manner in which connections are made among genes within the subnetwork determine the structure of the process that underlies the repertoire of functions the network is capable of displaying independent of its context. It is from this perspective that we first motivate our model, by describing a simple representation of observing gene expression in a manner capable of mimicing the embedding of a given subnetwork within any possible network context.

\ref{fig:expression_concept}A shows a simplified representation of a bacterial chromosome or plasmid encoding a small gene regulatory network (GRN). The amount of a given transcript present in a cell given in terms of the discrete counts obtained via sequence census methods (e.g. RNA-seq) or relative abundance derived from microarray data can be binned into a smaller number of discrete classes by setting a collection of thresholds on the original data set. If only a single threshold is given, then the data can be binned into two classes depending upon whether or not the original measurement surpasses the given threshold \ref{fig:expression_concept}B. If a large enough number of thresholds is available to distinguish among all possible molecule counts, then this observational protocol becomes complementary to mechanistic models of stochastic gene expression that seek solutions of the master equation, which contain probability distributions over molecule counts \cite{Walczak2009,Mugler2009}. At the lowest-level of the genotype-phenotype mapping, these coarse-grained expression values can be viewed as collectively determining the lowest level in the aforementioned hierarchy of phenotypes. In this way, the data derived from a number of samples of a given network can be described using binary sequences \ref{fig:expression_concept}C. For a given number of genes, it is possible to sample any subset of genes and the structure of this sampling process plays a role in determining the interpretation of the structure of the underlying process that generates the data, which, in this case, includes all regulatory elements that impinge upon the GRN module under consideration. For example, if, whenever we sample, we always consider the expression state of three out of three genes, we imply that the underlying process can be represented as a joint probability distribution over those three genes \ref{fig:expression_concept}C top. If instead we consider each pair of genes separately \ref{fig:expression_concept}, we imply the existence of a different kind of structure, which is the sum or union of several simpler spaces of probability distributions, each representing lower order interactions within the network that has been conceptually carved out of a larger network context.

\subsection{Divergence between observational structure and the structure of the underlying process driving gene expression can lead to inconsistency}
We can use an undirected graph (e.g. \ref{fig:inconsistentthreecycle}A, bottom left) to indicate interactions allowed by various GRN architectures (GRNAs). Each node of the graph is associated to the probability distribution that specifies probabilities for each gene to be observed in each of the states specified by the coarse-graining process described in the previous section. Each edge of the graph specifies a joint probability distribution for both of the nodes it contains (or connects) to simultaneously take on a given pair of values. In this context it is presumed that via marginalization the probability distributions associated to each edge are consistent with the probability distributions associated to each node. This description is similar to that of a Markov Random Field (a particular type of graphical model defined using undirected graphs) as a model of the stationary distribution of the stochastic process underlying gene expression \cite{Lauritzen1996,Chen2013a}. However, the implication of existence of a factorizable joint probability distribution over all nodes in the graph, which is implicit in the machine learning interpretation of graphical models, is not assumed here (Supplementary Information) \cite{Barber2012,Bishop2007,Murphy2012,Koller2009}.

\ref{fig:inconsistentthreecycle}A provides a concrete example where we assume three genes interact via all possible pairwise combinations and that via the coarse-graining process we have binned the expression state of each gene into one of two classes. Note that this graphical model is completely general and we can have, in principle, an arbitrary number of genes and an arbitrary number of classes in the assignment of expression states up to the maximal resolution of the method by which the observations are performed. Each node of the graph in \ref{fig:inconsistentthreecycle}A represents a probability distribution over the observation of each gene in either of the two states established in the coarse-graining process, which for node three are given by $p((3 \mapsto 0, 3 \mapsto 1))=(0.5,0.5)$. Given an edge we consider a map or function that takes as input the two genes connected by or contained within that edge and maps those two genes simultaneously to the available expression states. For example in \ref{fig:inconsistentthreecycle}A bottom there are maps $12 \mapsto 11$, $23 \mapsto 10$, and $31 \mapsto 10$ with associated probabilities $p(12 \mapsto 11)=0.1$ \ref{fig:inconsistentthreecycle}A top purple box and bottom purple arrows, $p(23 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top orange box and bottom orange arrows, and $p(31 \mapsto 10)=0.1$ \ref{fig:inconsistentthreecycle}A top green box and bottom green arrows. Each of the probability tables adjacent to each edge in the graph in \ref{fig:inconsistentthreecycle}A assigns a probability distribution to a set of the maps from the nodes connected by the edge to all possible combinations of the expression states. As these maps take collections of genes as input and produce collections of expression states as outputs we refer to them as genotype-phenotype mappings and the associated probability distributions are thus probability distributions over genotype-phenotype maps.

If the probability distribution in \ref{fig:inconsistentthreecycle}A represents the actual structure and particular parameters of the process generating the gene expression data, we can also consider the problem of inferring the parameters of a hierarchical model (Supplementary Information) for the probability distribution specified adjacent to the nodes and edges of the graph given the assumption that the structure of a given GRN module takes on the architecture of the given graph and then observe the GRN according to that assumption. \ref{fig:inconsistentthreecycle}B top left represents data containing three hundred observations generated by a process whose structure is equivalent to that of \ref{fig:inconsistentthreecycle}A except that we consider the case in which each expression state is equally likely. We now use maximum likelihood estimation \cite{Barber2012} to detertmine the parameters of a model like that of \ref{fig:inconsistentthreecycle}A given the dataset in \ref{fig:inconsistentthreecycle}B top left. \ref{fig:inconsistentthreecycle}B top middle shows that indeed all possible binary sequences representing the expression states for genes one, two and three are approximately uniformly represented in the given samples and the green bars in \ref{fig:inconsistentthreecycle}B bottom show the marginals of this joint distribution. Alternatively, we can infer the marginal probabilities using the sum-product (belief propagation) algorithm \cite{Barber2012} resulting in the yellow bars in \ref{fig:inconsistentthreecycle}B bottom. If we perform the same sequence of steps on the data in \ref{fig:inconsistentthreecycle}C top left, only the (loopy) belief propagation algorithm returns marginal distributions (\ref{fig:inconsistentthreecycle}C bottom) that are consistent with the data. This is a result of the fact that no joint probability distribution having these empirical marginals exists and maximum likelihood estimation determines a joint distribution that only approximates the empirical marginals. Dually, only the data of \ref{fig:expression_concept}C bottom and not that of \ref{fig:expression_concept}C top is consistent with the probabilities represented in \ref{fig:inconsistentthreecycle}A.

This inconsistency is a result of the fact that the network architecture in \ref{fig:inconsistentthreecycle}A contains a cycle \cite{Lauritzen1996,Geiger2006,Wainwright2007} and that we have given an ostensible data set and thus parameters that lead to the impossibility of constructing a consistent joint probability distribution over all three genes. For the case of the architecture in \ref{fig:inconsistentthreecycle}A, and moreover for any GRNA of any size that contains one or more cycles, the possibility of finding a joint distribution over all genes requires the implicit assumption that the structure of the generating process can be viewed simultaneously as that of \ref{fig:expression_concept}C top and that of \ref{fig:expression_concept}C bottom. The spaces of probability distributions associated to the two GRNAs contrasted in \ref{fig:expression_concept}C are different. This situation leads to the question of whether or not it is possible to classify the geometries and thus relationships among the spaces of probability distributions associated to all possible GRNAs on a given number of genes.

\subsection{Gene-regulatory network architecture is associated to the geometry of spaces of probability distributions over genotype-phenotype maps}
The collection of all possible GRNAs in the context of hierarchical models is equivalent to the lattice of all possible reduced subsets of genes (Supplementary Information). For example, \ref{fig:conediagram}A contains the lattice of reduced subsets of three genes. We are only interested in those subsets that contain at least one copy of each gene, which corresponds to the region highlighted with a gray background in \ref{fig:conediagram}A. Each GRNA induces a different structure on the genotype-phenotype maps underlying it. For example, \ref{fig:conediagram}B shows in the same vertical order the different structures induced by the three architectures highlighted in green in \ref{fig:conediagram}A.

We consider those GRNAs found lower in the lattice of \ref{fig:conediagram}A to be of higher modularity because each corresponds to the increasing restriction from higher- to lower-order interactions among genes and thus the restriction of simultaneous evaluation of phenotypes over the collection of genes. \ref{fig:conediagram}B top corresponds to the least modular GRNA because phenotype values are assigned simultaneously to all three genes. \ref{fig:conediagram}B middle exhibits an elevated degree of modularity because phenotypes are only simultaneously assigned to pairs of genes despite the fact that on timescales longer than that fundamental to the generating process it is possible to measure the phenotype values of all three genes simultaneously. Similarly, \ref{fig:conediagram}B bottom is even more modular because phenotype values are only assigned to one gene at a time. The structure of these GRNAs is associated to coverings of genotype space in the Supplementary Information.

Each of the GRNAs in \ref{fig:conediagram}A can be associated to a space of probability distributions over genotype-phenotype mappings. \ref{fig:conediagram}C schematically depicts the relationships among the probability distributions associated to the corresponding architectures and genotype-phenotype mappings in \ref{fig:conediagram}B. The inconsistency noted in the previous section between the architectures \ref{fig:conediagram}B top and middle is a result of the differing geometries in \ref{fig:conediagram}C middle. There, the smaller darker gray region corresponds to the space of all possible probability distributions for all possible parameters achievable via application of the linear transformation corresponding to marginalization to the probability distribution defined over all possible genotype-phenotype mappings associated to the network architecture in \ref{fig:conediagram}B top. Similarly, the lighter gray region corresponds to the analogous structure for \ref{fig:conediagram}B middle.

This relationship indicates that there are certain cases in which a given GRNA exhibiting a relatively higher degree of modularity nevertheless provides access to a larger space of possible probability distributions and thus of correlations among genes which can be coarsely referred to as gene expression patterns. These patterns serve as substrate for natural selection of GRNAs with particular functions by the network context within which the small modules we have considered so far function.

\subsection{Some GRNAs provide access to different collections of gene expression correlations}
Relationships between spaces of potential gene expression patterns like that of \ref{fig:conediagram}C middle occur for all GRNAs on any number of genes so long as there exists at least one cycle in the corresponding network architecture. For the case of three genes, there is only one class of graphs containing a cycle, which is that of \ref{fig:conediagram}B middle. For the case of four genes there are nine different classes of graphs containing cycles and these nine classes can be split into two groups depending upon whether or not the edges of the graphs are each restricted to contain only two genes (2-uniform hypergraphs, which are equivalent to undirected graphs). \ref{fig:non2uniformcyclichypergraphhasse} shows the componenets of the lattice analogous to that of \ref{fig:conediagram}A containing the five out of the nine isomorphism classes of hypergraphs on four genes that have cycles but that are not 2-uniform (the hypergraphs where at least one edge has more than two genes).

Given this larger collection of GRNAs with cycles we can assess the relative sizes of the analogs to the light and dark gray spaces (\ref{fig:conediagram}C middle) of probability distributions over genotype-phenotype maps. We assess the likelihood of choosing a point in the dark gray region at random by computing the ratio of the volume of the dark gray region (associated to the non-modular GRNA analogous to that of \ref{fig:conediagram}B top with a single edge containing all four genes), whose architecture and thus volume is fixed, to that of the light gray region, whose volume varies according to each of the cyclic graphs associated to a GRNA on four genes. We refer to this number as the non-modular:modular volume ratio or~$\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}$~(Supplementary Information). \ref{fig:ncycvolrat}A shows the relevant computations for seven different graphs where the edges are restricted to each contain two genes. The spaces are equivalent and thus the volume ratio equal to one for graphs lacking cycles (e.g. the first three graphs along the $x$-axis of \ref{fig:ncycvolrat}A). For the four network architectures in \ref{fig:ncycvolrat}A containing cycles, the volume ratio is consistently less than one. This quantifies the degree to which the space of probability distributions associated to the network architecture depicted along the $x$-axis is larger than the space of probability distributions associated to the non-modular architecture.

\subsection{Over evolutionary timescales network context can impose constraints on network architecture via the requirement of particular expression patterns}

The difference in the repertoires of gene expression patterns available to the various architectures with cyclic components each with respect to the structure corresponding to the joint distribution over all genes is equivalent to the statement that certain probability distributions over genotype-phenotype maps are only accessible from a certain class of GRNAs. This indicates that if the context (see \ref{fig:stochdynscheme} left) within which the network modules we have considered so far are embedded places certain functional requirements upon them, then the ability of any GRN to be able to meet those requirements must be considered relative to the geometry of its associated space of probability distributions. The ability to access certain correlation patterns is also dependent upon the network context.

For those GRNAs containing cycles, there are certain functional requirements that can be achieved so long as only local and not global consistency is required of them (Supplementary Information). Once global consistency is imposed as in the structure corresponding to the joint distribution over all genes, those functions that were accessible when only local consistency is imposed will become unavailable. This observation suggests a mechanism by which the GRNA may be imposed upon a collection of genes. \ref{fig:stochdynscheme} right shows a schematic of one potential scenario. The black dots in the center represent an initial condition of a stochastic process that is selected for its ability to achieve one of two different stationary distributions represented by the blue and the red dots respectively. The architecture whose space of probability distributions is represented in the top row of \ref{fig:stochdynscheme} is however unable to precisely achieve as its stationary distribution the point represented by the red dot in the bottom row of \ref{fig:stochdynscheme}.

When selective pressure is induced equivalent to the distribution located at the blue dot, either of the architectures are essentially equivalent with respect to the statistics of samples from their corresponding probability distributions and they can thus be considered as members of an evolutionarily neutral space. On the other hand, selective pressure equivalent to the collection of probability distributions located at the red point differentiates between the networks of the top and bottom row or equivalently between the network of the bottom row when global consistency is imposed versus the same network when only local consistency conditions are imposed. The same qualitative relationship holds true for the spaces of probability distributions of all GRNAs of any size and for any number of different expression levels so long as the graph associated to the space of probability distributions contains at least one cycle.
