%!TEX root = ../plos_template.tex
\subsection{Coarse-graining gene expression admits a simple probabilistic model}
\ref{fig:expression_concept}A shows a simplified representation of a bacterial chromosome or plasmid encoding a small gene regulatory network (GRN) the internal interactions of which constitute a GRN architecture (GRNA). The amount of a transcript present in a cell given in terms of the discrete counts obtained via sequence census methods like RNA-seq or relative abundance derived from microarray data can be binned into a smaller number of discrete classes by setting a collection of thresholds on the original data set. If only a single threshold is given, then the data can be binned into two classes depending upon whether or not the original measurement surpasses the given threshold \ref{fig:expression_concept}B. If a large enough number of thresholds is available to distinguish among all possible molecule counts, then this observational protocol serves as a statistical model complementary to mechanistic models of stochastic gene expression that seek solutions of the master equation, which contain probability distributions over molecule counts \cite{Walczak2009,Mugler2009}. At the lowest-level of the genotype-phenotype mapping, these coarse-grained expression values can be viewed as collectively determining the first level in the aforementioned hierarchy of genotype-phenotype maps. In this way, the data derived from a number of samples of a given network can be described using binary sequences that reflect time-lagged promoter states in addition to representing gene expression levels \ref{fig:expression_concept}C. For a given number of genes, it is possible to sample any subset of genes and the structure of this sampling process contributes to the interpretation of the structure of the underlying process that generates the data, which, in this case, includes all regulatory elements that impinge upon the GRNA under consideration. For example, if, whenever we sample, we always consider the expression state of three out of three genes, we imply that the underlying process can be represented by a joint probability distribution over the expression states of those three genes \ref{fig:expression_concept}C top. If instead we consider each of the three possible pairs of genes separately \ref{fig:expression_concept}, the existence of a different structure is implied, which is the sum or union of several simpler spaces of probability distributions, each representing lower order interactions within the network. What is most crucial in order to investigate the existence and maintenance of a given GRNA is the structure of the interactions between that GRNA and the network context in which it is embedded because the sampling process performed by the network context is what evaluates the function of a subnetwork and thereby determines the direction in which natural selection acts upon its architecture.

\subsection{Divergence between gene regulatory network architecture and the structure of connections to the network context can lead to inconsistency}
We can use an undirected graph (e.g. \ref{fig:inconsistentthreecycle}A, bottom left) to indicate interactions allowed by various GRNAs. Each node of the graph is associated to the probability distribution that specifies probabilities for each gene to be observed in each of the states specified by the coarse-graining process described in the previous section. Each edge of the graph specifies a joint probability distribution for both of the nodes it contains (or connects) to take on a given pair of values. We generalize this in what proceeds to hypergraphs where each edge can contain an arbitrary number of nodes. In this context it is presumed that via marginalization the probability distributions associated to each edge are consistent with the probability distributions associated to each node.

Inconsistency can result from network architectures containing cycles such as that in \ref{fig:inconsistentthreecycle}A \cite{Lauritzen1996,Geiger2006,Wainwright2007}. In this case it is possible to place functional requirements in terms of expression state probabilities upon a network that lead to the impossibility of constructing a consistent joint probability distribution over all genes in the network. For the case of the architecture in \ref{fig:inconsistentthreecycle}A, and moreover for any GRNA of any size that contains one or more cycles, the possibility of finding a joint distribution over all genes requires the implicit assumption that the structure of the generating process can be viewed simultaneously as that of \ref{fig:expression_concept}C top and that of \ref{fig:expression_concept}C bottom. The spaces of probability distributions associated to the two GRNAs contrasted in \ref{fig:expression_concept}C are different (Supplementary Material). This situation leads to the question of whether or not it is possible to classify the geometries and thus relationships among the spaces of probability distributions associated to all possible GRNAs on a given number of genes.

\subsection{Gene-regulatory network architecture is associated to the geometry of spaces of probability distributions over genotype-phenotype maps}
The collection of all possible GRNAs in our model is equivalent to the lattice of all possible reduced subsets of genes (Supplementary Material). For example, \ref{fig:conediagram}A contains the lattice of reduced subsets of three genes. We are only interested in those subsets that contain at least one copy of each gene, which corresponds to the region highlighted with a gray background in \ref{fig:conediagram}A. Each GRNA induces a different structure on the genotype-phenotype maps underlying it. For example, \ref{fig:conediagram}B shows in the same vertical order the different structures induced by the three architectures highlighted in green in \ref{fig:conediagram}A.

We consider those GRNAs found lower in the lattice of \ref{fig:conediagram}A to be of higher modularity because each corresponds to the restriction from higher- to lower-order genetic interactions relevant to the determination of phenotypes evaluated by a given network context. \ref{fig:conediagram}B top corresponds to the least modular GRNA because phenotype values are determined by interaction among all three genes. \ref{fig:conediagram}B middle exhibits an elevated degree of modularity because phenotypes are determined by pairwise interactions among genes. \ref{fig:conediagram}B bottom is even more modular because phenotype values are determined by each gene independently.

Each of the GRNAs in \ref{fig:conediagram}A can be associated to a space of probability distributions over genotype-phenotype mappings. \ref{fig:conediagram}C schematically depicts the relationships among the probability distributions associated to the corresponding architectures and genotype-phenotype mappings in \ref{fig:conediagram}B. The potential inconsistency noted in the previous section between the architectures \ref{fig:conediagram}B top and middle is a result of the differing geometries in \ref{fig:conediagram}C middle. There, the smaller darker gray region corresponds to the space of all possible probability distributions for all possible parameters achievable via application of the linear transformation corresponding to marginalization to the probability distribution defined over all possible genotype-phenotype mappings associated to the network architecture in \ref{fig:conediagram}B top. Similarly, the lighter gray region corresponds to the analogous structure for \ref{fig:conediagram}B middle.

This relationship indicates that there are certain cases in which a given GRNA exhibiting a relatively higher degree of modularity provides access to a larger space of extrinsically selectable correlations among genes. These gene expression patterns may serve as substrate for natural selection of GRNAs resulting from the imposition of functional requirements by the network contexts within which the modules we have considered so far are embedded.

\subsection{Quantifying relationships among selectable collections of gene expression correlations associated to different gene-regulatory network architectures}
Relationships between spaces of potential gene expression patterns like that of \ref{fig:conediagram}C middle occur for all GRNAs on any number of genes so long as there exists at least one cycle in the corresponding graph that represents the network architecture. For the case of three genes, there is only one class of graphs containing a cycle, which is that of \ref{fig:conediagram}B middle. For the case of four genes there are nine different classes of graphs containing cycles and these nine classes can be split into two groups depending upon whether or not the edges of the graphs are each restricted to contain only two genes (2-uniform hypergraphs, which are equivalent to undirected graphs). \ref{fig:non2uniformcyclichypergraphhasse} shows the components of the lattice analogous to that of \ref{fig:conediagram}A containing the five out of the nine isomorphism classes of hypergraphs on four genes that have cycles but that are not 2-uniform (the hypergraphs where at least one edge has more than two genes).

Given this larger collection of GRNAs with cycles we can assess the relative sizes of the analogs to the light and dark gray spaces (\ref{fig:conediagram}C middle) of probability distributions over genotype-phenotype maps. We assess the likelihood of choosing a point in the dark gray region at random by computing the ratio of the volume of the dark gray region (associated to the non-modular GRNA analogous to that of \ref{fig:conediagram}B top with a single edge containing all four genes), whose architecture and thus volume is fixed, to that of the light gray region, whose volume varies according to each of the cyclic graphs associated to a GRNA on four genes. We refer to this number as the non-modular:modular volume ratio or~$\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}$~(Supplementary Material). \ref{fig:ncycvolrat}A shows the relevant computations for seven different graphs where the edges are restricted to each contain two genes. The spaces are equivalent and thus the volume ratio equal to one for graphs lacking cycles (e.g. the first three graphs along the $x$-axis of \ref{fig:ncycvolrat}A). For the four network architectures in \ref{fig:ncycvolrat}A containing cycles, the volume ratio is consistently less than one. This quantifies the degree to which the space of probability distributions associated to the network architecture depicted along the $x$-axis is larger than the space of probability distributions associated to the non-modular architecture.

\subsection{Over evolutionary timescales network context can impose constraints on network architecture via the requirement of particular expression patterns}

The difference in the repertoires of gene expression patterns available to the various architectures with cyclic components each with respect to the structure corresponding to the joint distribution over all genes is equivalent to the statement that certain probability distributions over genotype-phenotype maps are only accessible from a certain class of GRNAs. This indicates that if the context (see \ref{fig:stochdynscheme} left) within which the network modules we have considered so far are embedded places certain functional requirements upon them, then the ability of any GRN to be able to meet those requirements must be considered relative to the geometry of its associated space of probability distributions. The ability to access certain correlation patterns is also dependent upon the connectivity between a given subnetwork and its network context.

For those GRNAs containing cycles, there are certain functional requirements that can be achieved so long as only local and not global consistency is required of them (Supplementary Material). Once global consistency is imposed as in the structure corresponding to the joint distribution over all genes, those functions that were accessible when only local consistency is imposed will become unavailable. This observation suggests a mechanism by which the GRNA may be imposed upon a collection of genes. \ref{fig:stochdynscheme} right shows a schematic of one potential scenario. The black dots in the center represent an initial condition of a stochastic process that is selected for its ability to achieve one of two different stationary distributions represented by the blue and the red dots respectively. The architecture whose space of probability distributions is represented in the top row of \ref{fig:stochdynscheme} is however unable to precisely achieve as its stationary distribution the point represented by the red dot in the bottom row of \ref{fig:stochdynscheme}.

When selective pressure is induced equivalent to the distribution located at the blue dot, either of the architectures are essentially equivalent with respect to the statistics of samples from their corresponding probability distributions and they can thus be considered as members of an evolutionarily neutral space. On the other hand, selective pressure equivalent to the collection of probability distributions located at the red point differentiates between the networks of the top and bottom row or equivalently between the network of the bottom row when global consistency is imposed versus the same network when only local consistency conditions are imposed. The same qualitative relationship holds true for the spaces of probability distributions of all GRNAs of any size and for any number of different expression levels so long as the graph associated to the space of probability distributions contains at least one cycle.
