%!TEX root = ../plos_template.tex

% \section{Hierarchical models}\label{sec:hierarchicalmodels}
% Given the collection of mappings $\mathcal{E}(O) = P^O$ for all $O \in \mathcal{G}$ we can attempt to explicitly construct a particular joint probability distribution $s \in \mathcal{D}_{\mathbb{R}}\mathcal{E}(L)$ on $\mathcal{E}(L)$. As noted above, in some cases such as that of \ref{fig:inconsistentthreecycle}A and C, such attempts may fail. This requires, first, a choice of mapping $\mathbb{R}^{\mathcal{E}(L)} \rightarrow \mathcal{D}_{\mathbb{R}}\mathcal{E}(L)$, which in the case of so-called \emph{exponential families} is
% \begin{eqnarray*}
% \text{exp} \colon \mathbb{R}^{\mathcal{E}(L)} &\rightarrow& \mathcal{D}_{\mathbb{R}}\mathcal{E}(L),\\
% g &\mapsto& \frac{e^g}{\sum_{d \in \mathcal{E}(L)} e^{g(d)}}.
% \end{eqnarray*}
% In this case the exponential family is the image $exp(\mathcal{I}) \subset \mathcal{D}_{\mathbb{R}}\mathcal{E}(L)$ where $\mathcal{I} \subseteq \mathbb{R}^{\mathcal{E}(L)}$.

% Given $\mathcal{G}$ we can associate an interaction subspace
% $$
% \mathcal{I}_{O} = \{ g \in \mathbb{R}^{\mathcal{E}(L)}\,\, |\,\, g(d_O,d_{L \setminus O}) = g(d_O,d'_{L \setminus O}) \text{ for all } d_O \in \mathcal{E}(O) \text{ and } d_{L \setminus O},d'_{L \setminus O} \in \mathcal{E}(L \setminus O)\}
% $$
% to each $O \in \mathcal{G}$. We can then construct the space of interactions associated to $\mathcal{G}$ itself as the sum over each interaction subspace
% $$
% \mathcal{I}_{\mathcal{G}} = \sum_{O \in \mathcal{G}} \mathcal{I}_{O}.
% $$
% The \emph{hierarchical model} \cite{Lauritzen1996} associated to $\mathcal{G}$ is then the exponential family
% $$
% \mathcal{E}xp_{\mathcal{G}} = \text{exp}(\mathcal{I}_{\mathcal{G}}).
% $$
% Markov random fields in the context of probabilistic graphical models or just graphical models are special cases of hierarchical models \cite{Lauritzen1996,Murphy2012,Barber2012,Bishop2007,Koller2009}. The description in this section is taken from \cite{Lauritzen1996}, which uses hypergraphs explicitly. In the main text, we use this notion in the numerical simulations in \ref{fig:inconsistentthreecycle}. Our reduced hypergraphs which correspond to GRNAs modeled as coverings of the collection of genes do not imply that they are instances of hierarchical models, but they can be taken to be a specification of the underlying hypergraph, also referred to as a \emph{factor graph}, of a hierarchical model.

\section{Example}\label{sec:example}
For the purposes of this example, we take the full set of genes to be $L = \{ l_1,l_2,l_3,l_4 \}$. Consider the case in which each of the gene regulatory network modules under consideration has two genes and we specify the covering of the genotype space given by $\mathcal{G} = \{\{l_1,l_2 \},\{l_1,l_4 \},\{l_3,l_2\},\{l_3,l_4\} \}$. The full set of genes together with a covering of the genotype space is a special case of a hypergraph $\mathcal{H} = (L,\mathcal{G})$ where the set of genes $L$ serve as the nodes or vertices and the covering $\mathcal{G}$ contains the edges or hyperedges, each of which is simply some subset of $L$. We also have two phenotype values $P = \{0, 1\}$. Each genotype-phenotype map in \ref{tab:gpm} assigns phenotype values to each $O \in \mathcal{G}$. For example, $e_1 \colon   l_1 l_2 \rightarrow 00$. \ref{tab:probabilities} likewise assigns probabilities to each of these \gnpm{}.

We can collect all of these \gnpm{} into one set using \ref{eq:gpfunctor} as $\mathcal{E}(\mathcal{G}) = P^{\mathcal{G}} = \{e_i | i=1 \ldots 16 \}$. The operation $\mathcal{D}_R$ given in \ref{eq:distfunctor} converts the set of all possible \gnpm{} into the set of probability distributions over that set of possible \gnpm{}. Successively applying these operations yields a distribution function $\mathcal{D}_R\mathcal{E}(\mathcal{G})=d \colon P^\mathcal{G} \rightarrow R$. If we now apply this distribution function $d$ to the set of \gnpm{} we get the probabilities associated to each genotype-phenotype map $d(\{e_i | i=1 \ldots 16 \}) = \{p_i|i=1 \ldots 16\}$ given in \ref{tab:probabilities}. This can also be written more abstractly as $\left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](\{e_i | i=1 \ldots 16 \}) = \left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](P^\mathcal{G}) = \left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](\mathcal{E}(\mathcal{G})) = \{p_i|i=1 \ldots 16\}$. Notice that we applied the operations $\mathcal{E}$ and $\mathcal{D}_R$ to a specific set, which was all of $\mathcal{G}$. We could have applied these operations to any $O \in \mathcal{G}$. In one case, for example if we take $O = \{l_1, l_2\}$, we would have gotten $\mathcal{E}(O) = \{e_i|i=1 \ldots 4\}$ and $\mathcal{D}_R\mathcal{E}(O) = d \colon P^O \rightarrow R$. In this case $\left[\mathcal{D}_R\mathcal{E}(O)\right](\mathcal{E}(O)) = \{p_i|i=1 \ldots 4\}$. We may even consider directly the meaning of $\left[\mathcal{D}_R\mathcal{E}(\{l_1\})\right](\mathcal{E}(\{l_1\}))$ and similarly for the other $l \in L$ for those subsets containing only one element of $L$.

Once we start considering probability distributions on subsets of $L$, we must introduce certain constraints to ensure that those probability distributions indeed derive from some probability distribution on the whole set $L$.  The sheaf condition then introduces a system of equations relating and thus constraining the probability parameters, beyond the constraint that each independently sum to one. The importance here of the sheaf condition is that any collection of data that might represent contingency tables or marginal distributions must satisfy certain conditions in order to be able to be derived via the standard marginalization procedure from some joint distribution. Without satisfying these conditions, one may just have an incoherent collection of observations that could not have derived from any joint probability distribution.  The constraint introduced by the sheaf condition is expressed in \ref{eq:sheafprob}, specialized to this example by
\begin{eqnarray}\label{eq:sheafprob2}
\sum_{e \in \mathcal{E}(\{l\}),\, e|l=e_{\{l\}}} d_{\{l\}}(e) \,\, = \sum_{e \in \mathcal{E}(O),\, e|l=e_{\{l\}}} d_O(e) \,\, = \sum_{e' \in \mathcal{E}(O'),\, e'|l=e_{\{l\}}} d_{O'}(e'),
\end{eqnarray}
for each $l \in L$, $e_{\{l\}} \in \mathcal{E}(\{l\})$, and $\{O \in \mathcal{G}|l=o, o \in O\}$. The resulting equations are explicitly indicated in like colors between the columns of \ref{tab:sheaf} :
\begin{equation}
\begin{aligned}\label{eq:pparsys}
p^*_1 &= p_1 + p_3 = p_5 + p_7, &
p^*_2 &= p_2 + p_4 = p_6 + p_8,\\
p^*_3 &= p_9 + p_{11} = p_{13} + p_{15},&
p^*_4 &= p_{10} + p_{12} = p_{14} + p_{16},\\
p^*_5 &= p_1 + p_2 = p_9 + p_{10},&
p^*_6 &= p_3 + p_4 = p_{11} + p_{12},\\
p^*_7 &= p_5 + p_6 = p_{13} + p_{14},&
p^*_8 &= p_7 + p_8 = p_{15} + p_{16}.
\end{aligned}
\end{equation}

Consider the meaning of the sheaf condition for $\mathcal{D}_R\mathcal{E}(L)$. $d \in \mathcal{D}_R\mathcal{E}(L)$ specifies a distribution whose domain consists of deterministic functions $\mathcal{E}(L) = P^L$ that map each gene independently, and independent of the fact that some combinations of genes  comprise gene regulatory network modules consisting of potentially interacting genes, to a phenotype. Since we have stated that $\mathcal{E}(L)$ satisfies the sheaf condition, for a given covering of the genotype space, $\mathcal{G}$, a globally compatible function $e \in  \mathcal{E}(L)$ restricts to $e|O$ for each $O \in \mathcal{G}$ such that there is a tuple of phenotypes assigned to genes that participate in overlapping gene regulatory network modules. Each $e \in P^L$ is associated to a trivial distribution $\delta_e \in \mathcal{D}_R\mathcal{E}(L)$ such that
\begin{eqnarray}
\delta_e(e') =
\begin{cases}
1, & e' = e,\\
0, & e' \neq e.
\end{cases}
\end{eqnarray}
For each gene regulatory network module $O$ we note that $\delta_e|O = \delta_{e|O}$. In this case, we can write the empirically determined distribution such as \ref{tab:probabilities} for a particular gene regulatory network module $d_O$ in terms of the distribution defined on the global set of genes restricted to a particular gene regulatory network module $d|O$:
\begin{equation}\label{eq:factordist}
\begin{split}
d_O(e) &= d|O(e)\\
&= \sum_{e' \in \mathcal{E}(L),e'|O=e} d(e')\\
&= \sum_{e' \in \mathcal{E}(L)} \delta_{e'|O}(e) \, d(e') = \sum_{e' \in \mathcal{E}(L)} \delta_{e'}|O(e) \, d(e')
\end{split}
\end{equation}
where
\begin{equation*}
\delta_{e'}|O(e) = \prod_{o \in O} \delta_{e'|\{o\}}(e|\{o\}).
\end{equation*}
We recover the empirically determined distribution as a sum over the deterministic \gnpm{} weighted by the probabilities of the distribution $d \in \mathcal{D}_R\mathcal{E}(L)$ defined over the full set of such deterministic mappings.

Thus, when there is a global section, given by $d \in \mathcal{D}_R\mathcal{E}(L)$, for an empirically determined distribution on \gnpm{}, for example that in \ref{tab:probabilities}, then it may be the case that a mechanism that stochastically selects among deterministic \gnpm{} can satisfy or generate an empirically determined distribution.

\section{Existence of a global section for an empirically determined distribution of \gnpm{}}\label{sec:globalsectionempirical}
Given an empirically determined distribution on the \gnpm{} process, it may be helpful to be able to classify whether or not there exists a global section satisfying the sheaf condition in order to determine the modularity requirements of the empirical distribution. If an empirically determined distribution does have a global section, then it is possible to achieve the same global biological function specified in the empirical distribution using a single gene regulatory network module that contains all of the genes in the given universal set of genes $L$ as is obtained using the particular cover of the genotype space $\mathcal{G}$ that is associated to the empirically determined distribution. If the empirically determined distribution does not have a global section satisfying the sheaf condition, then there is some contingency regarding interactions between genes necessitating decomposition into smaller submodules than is represented by the whole of the universal set of genes $L$.

% For a cover of the genotype space, $\mathcal{G}$, we can construct an incidence matrix, $\mathbf{G}$, representing the relationship between \gnpm{} having as domain particular gene regulatory network module contexts given by the $O \in \mathcal{G}$ and those global \gnpm{} defined on $L$. This is equivalent to the more formal statement that there is some binary relation,
% \begin{equation}
% R \subseteq \mathcal{E}(L) \times \coprod_{O \in \mathcal{G}} \mathcal{E}(O),
% \end{equation}
% whose logical matrix representation, $\mathbf{G}$, we would like to construct. In the first case, $\mathcal{E}(L) = P^L$ we have $\{t_1, \ldots t_m\} = \{t_j | t_j \in P^L\}$. In the second case, $\mathcal{D}( \mathcal{E}(\mathcal{G})) = \coprod_{O \in \mathcal{G}} \mathcal{E}(O)$, we have $\{e_1, \ldots e_n\} = \coprod_{O \in \mathcal{G}} \mathcal{E}(O)$. So we have two sets of maps, one defined on $P^L$ and the other defined on $\mathcal{E}(O) = P^O$ for each $O \in \mathcal{G}$. This yields a method of specifying the intended relationship that defines $\mathbf{G}_{n \times m}$ in terms of $t_j \in P^L$ and $e_i \in \mathcal{E}(O)$ :
% \begin{eqnarray}
% \mathbf{G}[i,j] =
% \begin{cases}
% 1, & t_j|O = e_i,\\
% 0, & \text{otherwise}.
% \end{cases}
% \end{eqnarray}
% This matrix can be viewed as an operator acting via matrix multiplication on distributions
% \begin{eqnarray*}
% \mathbf{G} \colon \mathcal{D}(\mathcal{E}(L)) &\rightarrow& \mathcal{D}( \mathcal{E}(\mathcal{G})),\\
% d &\mapsto& (d|O)_{O \in \mathcal{G}},
% \end{eqnarray*}
% and thereby taking a global distribution defined on \gnpm{} whose domain is the full set of genes $L$ to the \emph{marginalized} distributions that are defined relative to gene regulatory network modules contained in a covering of the genotype space $\mathcal{G}$. This is to say that $\mathbf{G} \cdot d = (d|O)_{O \in \mathcal{G}} = \{d_O\}_{O \in \mathcal{G}}$. $\mathbf{G}$ provides a way of determining the kinds of environmentally determined distributions on \gnpm{} (i.e. $\{d_O\}_{O \in \mathcal{G}}$, \ref{tab:probabilities}) that can be derived from distributions (i.e. $\mathcal{D} ( \mathcal{E}(L) )$) defined on the global \gnpm{} (i.e. $\mathcal{E}(L)$ as opposed to $\mathcal{E}(O)$).

If we have an experimentally determined distribution on genotype phenotype maps in different genetic contexts (e.g. in \ref{tab:probabilities}) that we refer to as $\{d_O\}_{O \in \mathcal{G}}$, then we have an assignment of a probability value in $R$ to each genotype phenotype map $e_i \in \mathcal{E}(O)$ in \ref{tab:gpm}. We can collect these probabilities into a vector $\mathbf{V}_{n \times 1}$ where we note that $n$ is the same as that used to specify the number of \gnpm{} in the coproduct over gene regulatory network modules in a given covering of the genotype space $\{e_1, \ldots e_n\} = \coprod_{O \in \mathcal{G}} \mathcal{E}(O)$. We can specify as unknown a distribution on the set of globally defined genotype phenotype maps and collect the associated probabilities in a vector $\mathbf{X}_{m \times 1}$ where there is a probability weight associated to each $t_j \in \mathcal{E}(L)$. Now we can attempt to determine the probability distribution on globally defined \gnpm{} $\mathcal{E}(L)$ by solving the linear system of equations
\begin{equation}
\begin{split}
\mathbf{G}_{n \times m} \mathbf{X}_{m \times 1} = \mathbf{V}_{n \times 1},\\
\sum_{i=1}^{m} \mathbf{X}_{i} = 1.
\end{split}
\end{equation}
The constraint that $\mathbf{X}$ sum to one in order to satisfy the definition of a probability distribution can be integrated into the matrix equation by adjoining a row containing ones to $\mathbf{G}_{n \times m}$ and a row containing a one to $\mathbf{V}_{n \times 1}$
\begin{equation}\label{eq:auglinsys}
\mathbf{G}_{(n + 1) \times m} \mathbf{X}_{m \times 1} = \mathbf{V}_{(n + 1) \times 1}.
\end{equation}
Solutions, if any, to \ref{eq:auglinsys} serve to exhibit a global section for the empirically determined distribution specified in $\mathbf{V}$.

% \section{Example global section determination}
% There are three parameters that determine the relative size of important sets for understanding the model at hand. $g$ represents the number of genetic loci, $a$ represents the number of different alleles that can occur at each locus, and $p$ represents the number of different phenotype values that each locus can take on. From these parameters, which we use together to classify models satisfying the structural constraints described thus far, we can derive additional quantities characterizing the model. For a system with parameters $(g,a,p)$ there are $a^g$ different gene regulatory network modules to consider and each may be associated with $p^g$ phenotypes. There are thus $(ap)^g$ different \gnpm{} to consider with respect to the gene regulatory network modules meaning that this is the size of the set $\coprod_{O \in \mathcal{G}} \mathcal{E}(O)$. The set of all possible genes $L$ has size $ga$ and since there are $p$ different phenotypes that each gene can be mapped to, there are $p^{ga}$ different globally defined \gnpm{} meaning that this is the size of the set $\mathcal{E}(L) = P^L$. Now we can state the size of the sets $\{t_1, \ldots t_m\} = \{t_j | t_j \in P^L\}$ and $\{e_1, \ldots e_n\} = \coprod_{O \in \mathcal{G}} \mathcal{E}(O)$ because we have explicit formulas for $n = (ap)^g$ and $m = p^{ga}$ in terms of the defining characteristics of a $(g,a,p)$ class of models. From the values of $n$ and $m$ we know the dimensions of the logical matrix representation of the relation $R$ as $\mathbf{G}_{(ap)^g \times p^{ga}}$.

% For the case $(g=2,a=2,p=2)$, \ref{tab:pars222} shows the relationship between the abstract notation used throughout the model description and the concrete parameters introduced in this example. Using the notation of \ref{tab:pars222}, we can also explicitly construct $\mathbf{G}_{(ap)^g \times p^{ga}}$ as in \ref{tab:logmat222}.

In terms of the probability values expressed in tables \ref{tab:hidvarmod} and \ref{tab:probabilities} the resulting system of equations $\mathbf{G}_{n \times m} \mathbf{X}_{m \times 1} = \mathbf{V}_{n \times 1}$ is
\begin{equation}
\begin{aligned}\label{eq:globsys}
p_1 &= q_1 + q_2 + q_3 + q_4 &
p_2 &= q_5 + q_6 + q_7 + q_8 \\
p_3 &= q_9 + q_{10} + q_{11} + q_{12} &
p_4 &= q_{13} + q_{14} + q_{15} + q_{16}\\
p_5 &= q_1 + q_3 + q_5 + q_7 &
p_6 &= q_2 + q_4 + q_6 + q_8 \\
p_7 &= q_9 + q_{11} + q_{13} + q_{15} &
p_8 &= q_{10} + q_{12} + q_{14} + q_{16}\\
p_9 &= q_1 + q_2 + q_9 + q_{10} &
p_{10} &= q_5 + q_6 + q_{13} + q_{14} \\
p_{11} &= q_3 + q_4 + q_{11} + q_{12} &
p_{12} &= q_7 + q_8 + q_{15} + q_{16}\\
p_{13} &= q_1 + q_5 + q_{9} + q_{13} &
p_{14} &= q_2 + q_6 + q_{10} + q_{14} \\
p_{15} &= q_3 + q_7 + q_{11} + q_{15} &
p_{16} &= q_4 + q_8 + q_{12} + q_{16}
\end{aligned}
\end{equation}
Adding the additional rows to $\mathbf{G}$ and $\mathbf{V}$ to ensure that $\sum_i q_i = 1$ gives the system $\mathbf{G}_{(n + 1) \times m} \mathbf{X}_{m \times 1} = \mathbf{V}_{(n + 1) \times 1}$.

\section{Geometric comparison of spaces of modular and non-modular distributions over \gnpm{}}\label{sec:computegeometrygpm}
We ultimately aim to determine the fraction of the space of all \gnpm{} with respect to a given covering of the genotype space $\mathcal{G}$ satisfying the sheaf condition and thereby posessing a global distribution. We first describe the linear transformations between global and local distributions, referred to as marginalization maps, over \gnpm{} in terms of the simple example for the case $L = \{ l_1,l_2,l_3,l_4 \}$, $\mathcal{G} = \{\{l_1,l_2 \},\{l_1,l_4 \},\{l_3,l_2\},\{l_3,l_4\} \}$, $P=\{0,1\}$ where the elements of $P^L$ index the columns and elements of $\coprod_{O \in \mathcal{G}} \mathcal{E} (O)$ index the rows of the matrix given in \ref{tab:logmat222}. This example generalizes immediately to arbitrary $L$, $\mathcal{G}$, and $P$. The primary geometric object of interest for constructing the spaces of local and global distributions over \gnpm{} is the \emph{probability simplex}:
$$
\Delta_{m-1} = \left\{ (p_1, \ldots , p_m) \in \mathbb{R}^m \colon \sum_{i=1}^m p_i = 1 \text{ and } p_j \geq 0 \text{ for all } j \right\}.
$$
In the example case we have considered the mapping
\begin{equation}
\begin{aligned}\label{eq:marginalmap}
\mathbf{G} &\colon \mathbb{R}^{16} \longrightarrow \mathbb{R}^{16}\\
  &\,\,\,\, \cup \,\,\,\,\,\,\,\,\,\,\,\,\,\,\,\,\, \cup\\
  &\,\,\,\, \Delta_{15} \longrightarrow \Delta_3^{\oplus 4}
\end{aligned}
\end{equation}
where $\Delta_{15}$ represents the space of global distributions one point of which is defined in \ref{tab:hidvarmod} and likewise $\Delta_3^{\oplus 4} = \Delta_3 \oplus \Delta_3 \oplus \Delta_3 \oplus \Delta_3$ represents the space of local distributions one point of which is defined in \ref{tab:probabilities}. Note that the codomain $\Delta_3^{\oplus 4}$ of $\mathbf{G}$ is determined by $\mathcal{G} = \{\{l_1,l_2 \},\{l_1,l_4 \},\{l_3,l_2\},\{l_3,l_4\} \}$ and $P=\{0,1\}$. There are four copies, hence the $\oplus 4$, of the $\Delta_{4-1}$ simplex: one for each gene regulatory network module in $\mathcal{G}$. Based upon the parameters utilized in equations \ref{eq:globsys} we can write elements of each of these spaces as $(q_1, \ldots, q_{16}) \in \Delta_{15}$ and $((p_1, \ldots , p_4), \ldots, (p_{13},\ldots,p_{16})) \in \Delta_3^{\oplus 4}$. The constraints necessary to define $\mathbf{G}$ on points have already been given in equations \ref{eq:globsys} whereby
\begin{align*}
(q_1, \ldots , q_{16}) \mapsto &((q_1+q_2+q_3+q_4, q_5+q_6+q_7+q_8, q_9+q_{10}+q_{11}+q_{12},q_{13}+q_{14}+q_{15}+q_{16}), \ldots \\
&,(q_1 + q_5 + q_9 + q_{13}, q_2 + q_6 + q_{10} + q_{14}, q_3 + q_7 + q_{11} + q_{15}, q_4 + q_8 + q_{12} + q_{16} ) ).
\end{align*}
An ostensible inverse to $\mathbf{G}$, $\mathbf{G}^{-1} \colon \Delta_3^{\oplus 4} \longrightarrow \Delta_{15}$, is underdetermined and can only be estimated in general (often via maximum entropy)
\begin{align*}
((p_1, \ldots , p_4), \ldots, (p_{13},\ldots,p_{16})) \mapsto & \,\,?.
\end{align*}
In order to estimate the quantity of interest we consider the relationship between two different convex polytopes. The first
$$
\mathbb{M}(\mathcal{G}) \equiv \{\mathbf{V} \in \mathbb{R}^{16} | \mathbf{G}\mathbf{X}=\mathbf{V} \text{ for all } \mathbf{X} \in \Delta_{15}\}
$$
is referred to as the marginal polytope with respect to a given covering of the genotype space $\mathcal{G}$ and represents the space of globally consistent marginal distributions that have a global section and thus, in this case, a joint probability distribution defined on the set of all genes $L$. The second
$$
\mathbb{L}(\mathcal{G}) \equiv \{\mathbf{V} \in \mathbb{R}^{16} | \text{the constraints in \ref{eq:pparsys} hold}\}
$$
is referred to as the locally consistent polytope. More generally for arbitrary $\mathcal{G}$, $L$, $P$, and associated $\mathbf{G}$
\begin{equation}
\begin{aligned}\label{eq:polytopes}
\mathbb{M}(\mathcal{G}) &\equiv \{\mathbf{V} \in \mathbb{R}^{n} | \mathbf{G}\mathbf{X}=\mathbf{V} \text{ for all } \mathbf{X} \in \Delta_{m-1}\}\\
\mathbb{L}(\mathcal{G}) &\equiv \{\mathbf{V} \in \mathbb{R}^{n} | \text{the constraints in \ref{eq:sheafprob} hold}\}
\end{aligned}
\end{equation}
In order to assess the likelihood, with respect to some GRNA $\mathcal{G}$, of the existence of a global section satisfying the sheaf constraints given in \ref{eq:sheafprob} we would like to compute the ratio of the volumes of these two polytopes
\begin{equation}\label{eq:volrat}
\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}
\end{equation}
this is the quantity plotted for various GRNAs $\mathcal{G}$ in \ref{fig:ncycvolrat}.

\section{Constraint deduction method to derive H-representation of $\mathbb{L}(\mathcal{G})$}\label{sec:constraintdeductionmethod}
In order to compute the quantity in \ref{eq:volrat} we use an algorithm which we merely outline here (code implementing this algorithm is made available via a virtual machine that can be reconstructed using the instructions available on \href{https://github.com/cameronraysmith/sep}{github}):
\begin{enumerate}
\item Compute (a basis for) the cokernel of $\mathcal{G}$ where $coker(\mathcal{G}) = ker(\mathcal{G}^T) \subseteq \Delta_3^{\oplus 4}$. The cokernel gives the obstructions to the system $\mathbf{G}\mathbf{X}=\mathbf{V}$ having a solution. In order to eliminate these obstructions constraints must be imposed on $\Delta_3^{\oplus 4}$ and these constraints are given precisely via annihilating the cokernel.
\item Use the constraints on $\Delta_3^{\oplus 4}$ from step 1 necessary for the system $\mathbf{G}\mathbf{X}=\mathbf{V}$ to have a solution to eliminate variables from the system of inequalities $\mathbf{V} \geq 0$ giving a half-space representation or H-representation of the polytope $\mathbb{L}(\mathcal{G})$. This can be used to compute $\text{Vol}(\mathbb{L}(\mathcal{G}))$.
\item Compute the vertices of $\mathbb{L}(\mathcal{G})$ from the H-representation determined in step 2 giving a vertex representation or V-representation of $\mathbb{L}(\mathcal{G})$.
\item Filter the non-integer rational vertices from the collection computed in step 3 to produce a corresponding V-representation of $\mathbb{M}(\mathcal{G})$ \cite{Wainwright2007} proposition 8.3.
\item Compute $\text{Vol}(\mathbb{M}(\mathcal{G}))$ from the V-represention of $\mathbb{M}(\mathcal{G})$.
\end{enumerate}
For standard computations on polytopes, we make use of the standard algorithms incorporated by the polymake project \cite{Gawrilow2000}. In some cases, the volume computation is too costly to perform exactly. In those cases we use the approximation given in \cite{Cousins}. In the next section we walk through some key components of the algorithm for the running example $\mathcal{G} = \{\{l_1,l_2 \},\{l_1,l_4 \},\{l_3,l_2\},\{l_3,l_4\} \}$ and $P=\{0,1\}$.

The equalities derived by computing the cokernel of the matrix $\mathcal{G}$ given in \ref{tab:logmat222} and adjoining rows that enforce the normalization of the marginal distributions are represented as a matrix in \ref{eq:eqmat}.
\begin{equation}\label{eq:eqmat}
\begin{aligned}
\begin{bmatrix}
  -1 & -1 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & -1 & -1 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  -1 & 0 & -1 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  0 & -1 & 0 & -1 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & -1 & 0 & -1 & 0 & 1 & 1 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & -1 & 0 & -1 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0\\
  -1 & -1 & -1 & -1 & 1 & 0 & 1 & 0 & 1 & 0 & 1 & 0 & -1 & 0 & 0 & 1 & 0\\
  1 & 1 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1\\
  0 & 0 & 0 & 0 & 1 & 1 & 1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 1 & 1 & 0 & 0 & 0 & 0 & 1\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 1 & 1 & 1\\
\end{bmatrix}
\end{aligned}
\end{equation}
The final column represents the right-hand side of each equality. It turns out all but one of the normalization conditions is linearly dependent with respect to the other equalities and so we can reduce this set of $7+4=11$ constraints to the $8$ represented again in matrix form in \ref{eq:kceqsrefa}.
\begin{equation}\label{eq:kceqsrefa}
\begin{aligned}
\begin{bmatrix}
  1 & 0 & 0 & -1 & 0 & 0 & 1 & 1 & 0 & 0 & 1 & 1 & 0 & 0 & 0 & 0 & 1\\
  0 & 1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & -1 & -1 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 1 & 1 & 0 & 0 & -1 & -1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 1\\
  0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & -1 & 0 & -1 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0 & 0 & 1 & 1 & 1\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0 & -1 & -1 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1 & 1 & 1 & 1\\
\end{bmatrix}
\end{aligned}
\end{equation}
These equalities can now be substituted into the positivity inequalities necessary to define any space of probability distributions. This yields a set of inequalities \ref{eq:polrepindineqs} that specify an H-representation of the polytope $\mathbb{L}(\mathcal{G})$. This is the modular polytope, which is a subspace of $\Delta_3^{\oplus 4}$ associated to distributions consistent with the linear transformation $\mathbf{G}$
\begin{equation}\label{eq:polrepindineqs}
\begin{aligned}
\begin{bmatrix}
  1 & 1 & -1 & -1 & -1 & -1 & 0 & 0 & 0\\
  0 & -1 & 0 & 0 & 1 & 1 & 0 & 0 & 0\\
  0 & -1 & 1 & 1 & 0 & 0 & 0 & 0 & 0\\
  1 & 0 & -1 & 0 & 0 & 0 & -1 & 0 & -1\\
  0 & 0 & 0 & -1 & 0 & 0 & 1 & 0 & 1\\
  1 & 0 & 0 & 0 & -1 & 0 & 0 & -1 & -1\\
  0 & 0 & 0 & 0 & 0 & -1 & 0 & 1 & 1\\
  1 & 0 & 0 & 0 & 0 & 0 & -1 & -1 & -1\\
  0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0\\
  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1\\
\end{bmatrix}
\end{aligned}
\end{equation}
A row $(a_0,a_1,...,a_d)$ corresponds to the inequality $a_0 + a_1 x_1 + ... + a_d x_d >= 0$. The embedded identity matrix has, in this particular case eight, rows that specify the positivity of the variables corresponding to each of the, in this particular case eight, dimensions. Transforming this inequality or H-representation to a vertex or V-representation of the modular polytope produces \ref{eq:vrepfromhrep}.
\begin{equation}\label{eq:vrepfromhrep}
\begin{aligned}
\begin{bmatrix}
  1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1\\
  1 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0\\
  1 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0\\
  1 & 1/2 & 1/2 & 0 & 1/2 & 0 & 1/2 & 1/2 & 0\\
  1 & 1/2 & 0 & 1/2 & 0 & 1/2 & 1/2 & 1/2 & 0\\
  1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
  1 & 1/2 & 1/2 & 0 & 0 & 1/2 & 0 & 0 & 1/2\\
  1 & 1/2 & 0 & 1/2 & 1/2 & 0 & 0 & 0 & 1/2\\
  1 & 0 & 0 & 1/2 & 0 & 1/2 & 0 & 0 & 1/2\\
  1 & 0 & 1/2 & 0 & 1/2 & 0 & 0 & 0 & 1/2\\
  1 & 0 & 1/2 & 0 & 0 & 1/2 & 1/2 & 1/2 & 0\\
  1 & 0 & 0 & 1/2 & 1/2 & 0 & 1/2 & 1/2 & 0\\
  1 & 1 & 1 & 0 & 1 & 0 & 0 & 0 & 0\\
  1 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0\\
  1 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0\\
  1 & 0 & 0 & 1 & 0 & 0 & 1 & 0 & 0\\
  1 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 0\\
  1 & 0 & 0 & 0 & 1 & 0 & 1 & 0 & 0\\
  1 & 0 & 1 & 0 & 0 & 0 & 0 & 1 & 0\\
  1 & 1 & 0 & 1 & 0 & 1 & 0 & 0 & 1\\
  1 & 1 & 0 & 1 & 1 & 0 & 1 & 0 & 0\\
  1 & 1 & 1 & 0 & 0 & 1 & 0 & 1 & 0\\
  1 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 1\\
  1 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 1\\
\end{bmatrix}
\end{aligned}
\end{equation}
This completes steps 1-3 of the algorithm outlined above. The remaining steps are trivial. For reference, they appear in the code in the Appendices that implements the algorithm for computing $\frac{\text{Vol}(\mathbb{M}(\mathcal{G}))}{\text{Vol}(\mathbb{L}(\mathcal{G}))}$ for arbitrary $\mathcal{G}$.

% \section{A generative process for apparently inconsistent pairwise distributions}\label{sec:generativeprocessapparentinconsistency}

% The inequalities defining $\mathbb{M}(\mathcal{G})$ were derived under the assumption that the two-element probabilities were obtained by mariginalizing a three-element distribution.  Hence, if some other procedure, such as conditionalization is used to obtain them instead, the inequalities need not apply, as in the example above, which may be described as follows:  Introduce the sample spaces
% \begin{align*}
% S_{1} &= \{e_0,e_1\} \\
% S_{2} &= \{f_0,f_1\} \\
% S_{3} &= \{g_0,g_1\} \\
% S_{12} &= \{b_{00},b_{01},b_{10},b_{11}\} \\
% S_{13} &= \{c_{00},c_{01},c_{10},c_{11}\} \\
% S_{23} &= \{d_{00},d_{01},d_{10},d_{11}\} \\
% S_{123} &= \{a_{002},a_{012},a_{102},a_{112},a_{020},a_{021},a_{120},a_{121},a_{200},a_{201},a_{210},a_{211}\}, \\
% \end{align*}
% the inclusion maps $\iota_1, \iota_2, \iota_3$ and the projections $\pi_1, \ldots. \pi_6$.
% $$
% \xymatrix{
% S_{1}& S_{13} \ar@{->>}[l]_{\pi_1} \ar@{->>}[r]^{\pi_2} \ar@{^{(}->}[d]_{\iota_2} & S_{3} \\
% S_{12} \ar@{->>}[u]^{\pi_3} \ar@{->>}[rd]_{\pi_5} \ar@{^{(}->}[r]^{\iota_1} & S_{123} & S_{23} \ar@{->>}[u]^{\pi_4} \ar@{->>}[ld]^{\pi_6} \ar@{_{(}->}[l]_{\iota_3} \\
% & S_2 & }
% $$
% \begin{center}
% \begin{tabular}{c | c}
% $x$ & $\iota_1 (x)$ \\ \hline
% $b_{00}$ & $a_{002}$ \\
% $b_{01}$ & $a_{012}$ \\
% $b_{10}$ & $a_{102}$ \\
% $b_{11}$ & $a_{112}$ \\
% \end{tabular} \qquad
% \begin{tabular}{c | c}
% $x$ & $\iota_2 (x)$ \\ \hline
% $c_{00}$ & $a_{020}$ \\
% $c_{01}$ & $a_{021}$ \\
% $c_{10}$ & $a_{120}$ \\
% $c_{11}$ & $a_{121}$ \\
% \end{tabular} \qquad
% \begin{tabular}{c | c}
% $x$ & $\iota_3 (x)$ \\ \hline
% $d_{00}$ & $a_{200}$ \\
% $d_{01}$ & $a_{201}$ \\
% $d_{10}$ & $a_{210}$ \\
% $d_{11}$ & $a_{211}$ \\
% \end{tabular}
% \end{center}

% \begin{center}
% \begin{tabular}{c | c}
% $x$ & $\pi_1 (x)$ \\ \hline
% $c_{00}$ & $e_0$ \\
% $c_{01}$ & $e_0$ \\
% $c_{10}$ & $e_1$ \\
% $c_{11}$ & $e_1$ \\
% \end{tabular} \quad
% \begin{tabular}{c | c}
% $x$ & $\pi_2 (x)$ \\ \hline
% $c_{00}$ & $g_0$ \\
% $c_{01}$ & $g_1$ \\
% $c_{10}$ & $g_0$ \\
% $c_{11}$ & $g_1$ \\
% \end{tabular} \quad
% \begin{tabular}{c | c}
% $x$ & $\pi_3 (x)$ \\ \hline
% $b_{00}$ & $e_0$ \\
% $b_{01}$ & $e_0$ \\
% $b_{10}$ & $e_1$ \\
% $b_{11}$ & $e_1$ \\
% \end{tabular} \quad
% \begin{tabular}{c | c}
% $x$ & $\pi_4 (x)$ \\ \hline
% $d_{00}$ & $g_0$ \\
% $d_{01}$ & $g_1$ \\
% $d_{10}$ & $g_0$ \\
% $d_{11}$ & $g_1$ \\
% \end{tabular} \quad
% \begin{tabular}{c | c}
% $x$ & $\pi_5 (x)$ \\ \hline
% $b_{00}$ & $f_0$ \\
% $b_{01}$ & $f_1$ \\
% $b_{10}$ & $f_0$ \\
% $b_{11}$ & $f_1$ \\
% \end{tabular} \quad
% \begin{tabular}{c | c}
% $x$ & $\pi_6 (x)$ \\ \hline
% $d_{00}$ & $f_0$ \\
% $d_{01}$ & $f_0$ \\
% $d_{10}$ & $f_1$ \\
% $d_{11}$ & $f_1$ \\
% \end{tabular}
% \end{center}
% \ref{fig:condmargprobspaces} shows these mappings in terms of colored subsets.

% It is thus possible to generate what appear to be inconsistent pairwise distributions by embedding the system in \ref{fig:inconsistentthreecycle} into a consistent system represented by \ref{fig:controlnetwork}. In this system we consider the collection of pairwise probability distributions $p(l_1,l_2)$, $p(l_2,l_3)$, $p(l_1,l_3)$, and the joint probability distribution $p(m_1,m_2,m_3,m_4)$ where the states of genes $l_i \in \{ 0, 1\}$ and genes $m_i \in \{0,1,2\} \equiv \mathcal{M}$. The random variables representing the states of genes $l_1,l_2,l_3$ correspond to $m_1,m_2,m_3$ respectively and $m_4$ is a control random variable colored in dark gray in \ref{fig:controlnetwork}. We specify the control exhibited by $m_4$ over $m_1,m_2,m_3$ via the constraints
% \begin{align*}
% p(m_1=2|m_4=0)=1,\\
% p(m_2=2|m_4=1)=1,\\
% p(m_3=2|m_4=2)=1.
% \end{align*}
% We set $p(l_1,l_2)$, $p(l_2,l_3)$, $p(l_1,l_3)$ equal to those along the respective edges in \ref{fig:inconsistentthreecycle}A and then enforce analogous constraints on $p(m_1,m_2,m_3,m_4)$
% \begin{align*}
% p(m_1,m_2|m_4=2)=p(l_1,l_2),\\
% p(m_2,m_3|m_4=0)=p(l_2,l_3),\\
% p(m_1,m_3|m_4=1)=p(l_1,l_3).
% \end{align*}
% Finally, we place a uniform distribution on the states of $m_4$, $\forall m \in \mathcal{M} \left[ p(m_4=m) = 1/|\mathcal{M}| \right]$. This leads to the joint distribution
% \begin{align*}
% p(m_1,m_2,m_3,m_4=\{0022\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{0122\})&=2/15,\\
% p(m_1,m_2,m_3,m_4=\{1022\})&=2/15,\\
% p(m_1,m_2,m_3,m_4=\{1122\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{0201\})&=2/15,\\
% p(m_1,m_2,m_3,m_4=\{0211\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{1201\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{1211\})&=2/15,\\
% p(m_1,m_2,m_3,m_4=\{2000\})&=2/15,\\
% p(m_1,m_2,m_3,m_4=\{2010\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{2100\})&=1/30,\\
% p(m_1,m_2,m_3,m_4=\{2110\})&=2/15,
% \end{align*}
% where probabilities for all unlisted states are precisely zero. Now if we compute the conditional distributions
% \begin{align*}
% p(m_1,m_2|m_4=2) = \begin{bmatrix}
% 0.1 & 0.4 & 0\\
% 0.4 & 0.1 & 0\\
% 0 & 0 & 0
% \end{bmatrix},\\
% p(m_2,m_3|m_4=0) = \begin{bmatrix}
% 0.4 & 0.1 & 0\\
% 0.1 & 0.4 & 0\\
% 0 & 0 & 0
% \end{bmatrix},\\
% p(m_1,m_3|m_4=1) = \begin{bmatrix}
% 0.4 & 0.1 & 0\\
% 0.1 & 0.4 & 0\\
% 0 & 0 & 0
% \end{bmatrix},
% \end{align*}
% we see that were we to only observe the variables $m_1,m_2,m_3$ we would arrive at pairwise distributions consistent with
% \begin{align*}
% p(l_1,l_2) = \begin{bmatrix}
% 0.1 & 0.4\\
% 0.4 & 0.1
% \end{bmatrix},\\
% p(l_2,l_3) = \begin{bmatrix}
% 0.4 & 0.1\\
% 0.1 & 0.4
% \end{bmatrix},\\
% p(l_1,l_3) = \begin{bmatrix}
% 0.4 & 0.1\\
% 0.1 & 0.4
% \end{bmatrix},
% \end{align*}
% which cannot derive as marginals from any $p(l_1,l_2,l_3)$.

% % 1--2 conditioned on x_3=3, x_4=3
% %     0.4000    0.1000         0
% %     0.1000    0.4000         0
% %          0         0         0

% % 2--3 conditioned on x_1=3, x_4=1
% %     0.4000    0.1000         0
% %     0.1000    0.4000         0
% %          0         0         0

% % 1--3 conditioned on x_2=3, x_4=2
% %     0.1000    0.4000         0
% %     0.4000    0.1000         0
% %          0         0         0

% This scenario may be implemented via a synthetic gene circuit with four genes, where each of the first three genes contains an operator site for each of the other two and all three genes contain an operator site for the fourth gene \ref{fig:condgenescenario}. If the fourth gene is constitutively expressed at a low rate so that it persists at low copy number it is uniformly, in time, likely to be bound to the operator site present on each of the first three genes. When gene four is bound to its operator site on any of the first three genes, that gene's state is $2$. If gene four is absent, the output state is determined by the presence or absence of the other two operator sites according to each of the tables depicted in \ref{fig:condgenescenario}.

\FloatBarrier
\pagebreak

\begin{figure}[!ht]
\centering
\noindent\includegraphics[width=0.8\columnwidth]{fig/efunctor.pdf}
\caption{{\bf Example of the functor mapping subsets of genes to measurable spaces.} (A) On the left hand side are subsets of $L=\{l_1,l_2\}$ ordered by inclusion. On the right hand side are the spaces of gene network-phenotype maps also ordered by inclusion. The labels for the maps define them. For example, $e^{12}_{01}(l_1) = 0$ and $e^{12}_{01}(l_2) = 1$. (B) For the given covering, $\mathcal{G}$, the associated marginalization matrix acting on the probability vector $\{ p^{12}_{00},p^{12}_{01},p^{12}_{10},p^{12}_{11} \}$ to give $\{ p^{1}_{0},p^{1}_{1},p^{2}_{0},p^{2}_{1} \}$ is $\mathbf{G}$.}
\label{fig:efunctor}
\end{figure}

\pagebreak

\begin{figure}[!ht]
\centering
\noindent\includegraphics[width=0.4\columnwidth]{fig/abstractroadmap.pdf}
\caption{{\bf Mathematical relationships defining the hierarchy of phenotypes via coarse-graining.} }
\label{fig:abstractroadmap}
\end{figure}

% \pagebreak

% \begin{figure}[!ht]
% \centering
% \noindent\includegraphics[width=0.7\columnwidth]{fig/genesallelesgrns.pdf}
% \caption{{\bf Sequences, partitions, and GRNs.} In the sequences panel, we consider the letters to be abstract labels for arbitrary genetic sequences that may or may not be found to have a particular function once genetic interactions are taken into account. Sequences may be grouped together, for example if they appear mutually exclusively in gene regulatory networks, under identifications such as~$A=X,\,B=X',\,C=Y,\,D=Y'$. Genomes could then, for example, take one gene from each subset of sequences, by analogy to the partitions representing alleles and the organisms being haploid with all genes are essential. However, in general, we may consider arbitrary subsets of genes to be able to be included in a genome.}
% \label{fig:genesallelesgrns}
% \end{figure}
% \begin{center}
% \includegraphics[width=0.9\columnwidth]{fig/genesallelesgrns.pdf}
% \end{center}
% \begin{figure}[!ht]
% \centering
% \noindent\includegraphics[width=0.4\columnwidth]{fig/potential-interactions.pdf}
% \includegraphics[width=0.4\columnwidth]{fig/two-gene-two-allele-interactions.pdf}
% \includegraphics[width=0.4\columnwidth]{fig/incompatibility-interactions.pdf}
% \includegraphics[width=0.4\columnwidth]{fig/dependency-interactions.pdf}
% \caption{{\bf Gene interaction lattices.}}
% \label{fig:geneinteractionlattices}
% \end{figure}
\pagebreak

\begin{figure}[!ht]
\centering
\noindent\includegraphics[width=1.0\columnwidth]{fig/non2uniformcyclichypergraphhasse.pdf}
\caption{{\bf Hierarchical relationships among all possible classes of hypergraphs that are not graphs (i.e. not 2-uniform) but have cycles.} (A) There is a Hasse diagram for the lattice of GRNAs analogous to that of \ref{fig:conediagram}A but defined on four rather than only three genes. Within this lattice some of the graphs have cycles and some do not. (B) The highest levels of the Hasse diagram associated to the lattice of GRNAs on four genes containing hypergraphs having cycles. (C) and (D) contain lower levels of GRNAs containing cycles. Each of the four panels in (D) are on the same level. In total, each level represents an isomorphism class of hypergraphs. Therefore, there are five isomorphism classes of non-2-uniform hypergraphs representing GRNAs on four genes that contain cycles leading to the relationship between spaces of probability distsributions on associated genotype-phentoype maps analogous to that of \ref{fig:conediagram}C.}
\label{fig:non2uniformcyclichypergraphhasse}
\end{figure}

\pagebreak

\begin{figure}[!ht]
\centering
\noindent\includegraphics[width=0.3\columnwidth]{fig/controlnetwork.pdf}
\caption{{\bf Embedding into a particular network context.} The network inside the blue circle is equivalent to that of \ref{fig:inconsistentthreecycle}A, embedded into a particular network context that provides inputs (dark gray node) and outputs (light gray nodes) to the focal subnetwork. As a result of their connectivity, the outputs depend here upon pairwise correlations within the three genes comprising the focal subnetwork.}
\label{fig:controlnetwork}
\end{figure}

% \pagebreak

% \begin{figure}[!ht]
% \centering
% \noindent\includegraphics[width=0.5\columnwidth]{fig/condmargprobspaces.pdf}
% \caption{{\bf Diagrammatic representation of the sample spaces for the process generating apparent inconsistency.}}
% \label{fig:condmargprobspaces}
% \end{figure}
