% Template for PLoS
% Version 1.0 January 2009
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template

\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

\usepackage{bussproofs}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

% xy-pic for diagrams
\usepackage[all]{xy}
% subcaption
\usepackage{subcaption}
% hyperref
\usepackage{hyperref}
% color table cells http://goo.gl/ZmpJv
\usepackage[table]{xcolor}

% Use the PLoS provided bibtex style
\bibliographystyle{plos2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

\pagestyle{myheadings}
%% ** EDIT HERE **


%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

%% END MACROS SECTION

\begin{document}

% Title must be 150 characters or less
\begin{flushleft}
{\Large
\textbf{Title}
}
% Insert Author names, affiliations and corresponding author email.
\\
Author1$^{1}$, 
Author2$^{2}$, 
Author3$^{3,\ast}$
\\
\bf{1} Author1 Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{2} Author2 Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{3} Author3 Dept/Program/Center, Institution Name, City, State, Country
\\
$\ast$ E-mail: Corresponding author@institute.edu
\end{flushleft}

% Please keep the abstract between 250 and 300 words
\section*{Abstract}

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLoS ONE authors please skip this step. 
% Author Summary not valid for PLoS ONE submissions.   
\section*{Author Summary}

\section*{Introduction}
In order to understand biological systems, it will be necessary to clarify the relationship between genotype and phenotype. The process of doing so may require revision to either or both concepts. Determining whether or not such revision is necessary will be facilitated by formalization of these concepts. This formalization should lay bare the assumptions implicit in the respective definitions of genotype and phenotype and allow for the graded incorporation of the current state of knowledge about the intermediate processes involved in constructing the genotype-phenotype relationship that have been gained through the interplay between theory and experiment since, at least, the advent of modern molecular biology.

Here we formalize the genotype-phenotype mapping. In doing so, we make use of a small fragment of the unifying mathematical language of category theory. We demonstrate the way in which expressing our results in this language facilitates expanding the domain of applicability of the model as well as connecting the model to what may seem \emph{a priori} as disparate scientific domains. In addition to the direct value of using category theory as a framework for organizing information for the purpose of constructing precise modeling descriptions, this approach suggests an efficient method of sharing modeling results between different fields of science.

% Results and Discussion can be combined.
\section*{Model description}

\subsection*{Deterministic genotype-phenotype maps}
Consider the case in which we have an arbitrary number of genetic loci indexed by $i=0,1,...$. There is a particular set of potential alleles denoted $L$. A subset of alleles will be denoted as $U \subseteq L$. We also have a set of phenotypes $P$. In principle, any allele could give rise to any phenotype. A gene regulatory network module, which in some cases may represent all of the genes in an organism, is represented by a tuple of alleles $O$ and a population is a set of such tuples $\mathcal{G}$ (i.e. $O \in \mathcal{G}$). A genotype-phenotype map is thus represented as a function mapping a subset of alleles into phenotypes 
$$
e \colon U \rightarrow  P.
$$

In the observationalist framework, our alleles $L$ are analogous to measurements $U \subseteq X$ and our phenotypes $P$ are like measurement outcomes or values $V$. An event in which outcomes $s(a)$ are observed for each $a \in U$ serve to define sections over $U$ as

$$
s \colon U \rightarrow V.
$$

In order to consider the set of all genotype-phenotype maps together, we can construct a functor

$$
\mathcal{E} \colon \mathcal{P}(L)^{opp} \rightarrow Set
$$
by exhibiting precisely how it acts on objects and morphisms in its domain category. We denote the set of all subsets of a set $X$ as $\mathcal{P}(X)$. Therefore $\mathcal{P}(L)$ is a category in which the objects are subsets of alleles and morphisms represent inclusion of a smaller subset into a larger superset (i.e. $U \subseteq U' \Rightarrow U \rightarrow U'$). The category $\mathcal{P}(L)^{opp}$ then has the same objects, but the morphisms represent restriction from a larger to a smaller subset of alleles (i.e. $U' \supseteq U \Rightarrow U' \rightarrow U$). We explicitly represent the way in which the functor $\mathcal{E}$ acts on objects and morphisms $U$ and $U \subseteq U'$ respectively as
\begin{equation}\label{eq:gpfunctor}
\begin{split}
\mathcal{E} \colon \mathcal{P}(L)^{opp} &\rightarrow Set,\\
U &\mapsto P^U,\\
U \subseteq U' &\mapsto res^{U'}_{U}.
\end{split}
\end{equation}
So the functor $\mathcal{E}$ takes a subset of alleles and returns the set of genotype-phenotype mappings from the given subset of alleles to the set of phenotypes. $\mathcal{E}$ is thus, by definition, a presheaf functor, which is an object in the functor category $Sets^{\mathcal{P}(L)^{opp}}$. The restriction map operates on sets deriving from the application of $\mathcal{E}$ to a subset of alleles as follows
\begin{eqnarray}
res^{U'}_{U} \colon \mathcal{E}(U') &\rightarrow& \mathcal{E}(U)\\
P^{U'} &\rightarrow& P^U\\
e' \colon U' \rightarrow P &\mapsto& e'|_U \colon U \rightarrow P
\end{eqnarray}

\subsection*{Stochasticity among genotype-phenotype maps}
There may be several sources for stochasticity in genotype-phenotype mapping including small numbers of the causal molecules and products of gene expression as well as environmental fluctuations upon which genotype-phenotype mappings are conditioned. Here we generalize the deterministic framework outlined above to the stochastic case.

An \emph{algebraic structure} is determined by a set and one or more finitary operations (e.g. binary multiplication) defined on the elements of that set. A \emph{monoid} is a type of algebraic structure determined by a set and a binary operation such that the latter satisfies closure, associativity and identity with respect to the given set. A \emph{semiring} is an algebraic structure determined by a set with two binary operations. One of the binary operations, addition, forms a commutative monoid and has identity element 0. The second binary operation, multiplication, is a monoid with identity element 1. These binary operations interact such that multiplication distributes over addition and multiplication by the identity element of the addition monoid annihilates all elements in the semiring. For example, the real numbers under addition and multiplication constitute a semiring whose data is described as $\left( \mathbb{R},+,0,\times,1 \right)$.

Consider a function, $\phi$ from a set, $L$, to a semiring $R$ written $\phi \colon L \rightarrow R$. The \emph{support} of $\phi$, $\text{supp}(\phi)$, is then the set $\{ l \in L | \phi(l) \neq 0 \}$. A distribution with respect to the semiring $R$ on $L$ is given as a function having finite support and satisfying a constraint
\begin{eqnarray}
d \colon L \rightarrow R,\\
\sum_{l \in L} d(x) = 1
\end{eqnarray}
We can consider the set of all distributions with respect to a given semiring $R$ satisfying the above constraints and defined on the set $L$ as being given by a functor applied to $L$ as $\mathcal{D}_R (L)$. We can again explicitly represent the way in which the functor $\mathcal{D}_R$ acts on objects and morphisms $L$ and $f \colon L \rightarrow M$ as
\begin{equation}\label{eq:distfunctor}
\begin{split}
\mathcal{D}_R \colon Set &\rightarrow Set,\\
L &\mapsto \mathcal{D}_R (L),\\
f \colon L \rightarrow M &\mapsto \mathcal{D}_R (f) \colon \mathcal{D}_R (L) \rightarrow \mathcal{D}_R (M),
\end{split}
\end{equation}

where
\begin{eqnarray}
\mathcal{D}_R (f) \colon \mathcal{D}_R (L) &\rightarrow& \mathcal{D}_R (M),\\
d &\mapsto& \left[ m \mapsto \sum_{f(l)=m} d(l) \right]. 
\end{eqnarray}

For the case in which we consider $R$ to be the semiring of non-negative real numbers $\left( \mathbb{R}_{\geq 0},+,0,\times,1 \right)$, $\mathcal{D}_R (L)$ represents the set of probability distributions on the set $L$.

Recalling the presheaf functor, $\mathcal{E} \colon \mathcal{P}(L)^{opp} \rightarrow Set$, mapping genotypes to the set of maps from those genotypes to the set of phenotypes, we can now compose it with the distribution functor $\mathcal{D}_R$ to obtain a new presheaf functor $\mathcal{D}_R \circ \mathcal{E} \colon \mathcal{P}(L)^{opp} \rightarrow Set \rightarrow Set$ that assigns to each genotype a distribution over the set of maps from those genotypes to the set of possible phenotypes. The action of $\mathcal{D}_R \mathcal{E}$ on objects and morphisms in $\mathcal{P}(L)^{opp}$ yields
\begin{eqnarray}
\mathcal{D}_R \mathcal{E} \colon \mathcal{P}(L)^{opp} &\rightarrow& Set,\\
U &\mapsto& \mathcal{D}_R \mathcal{E}(U) \equiv d \colon P^U \rightarrow R,\\ 
U \subseteq U' &\mapsto& \mathcal{D}_R \mathcal{E}(U') \rightarrow \mathcal{D}_R \mathcal{E}(U).
\end{eqnarray}
where
\begin{eqnarray}
\mathcal{D}_R \mathcal{E}(U') &\rightarrow& \mathcal{D}_R \mathcal{E}(U),\\
d \colon P^{U'} \rightarrow R &\rightarrow& d|U \colon P^{U} \rightarrow R
\end{eqnarray}
and
\begin{eqnarray}
d \colon P^{U'} &\rightarrow& R,\\
s' &\mapsto& d(s');\\
d|U \colon P^{U} &\rightarrow& R,\\
s &\mapsto& \sum_{s' \in \mathcal{E}(U'),\, s'|U=s} d(s').
\end{eqnarray}

\subsection*{Coverings of genotype space}
An individual may be composed of some subset of possible alleles where the set of all alleles, $L$, is viewed as a genotype space. A \emph{covering} of the genotype space, $\mathcal{G}$, satisfies $\cup_i \mathcal{G}_i = L$ and $O,O' \in \mathcal{G}$ and $O \subseteq O'$ means that $O = O'$. The first condition means that $\mathcal{G}$ covers $L$ and the second conditions means that if a given subset of alleles $O'$ is compatible in a sense to be explained more precisely in what proceeds then any smaller subset of alleles $O$ is also compatible.

\subsection*{Compatibility of distributions on genotype-phenotype maps}
Given a covering of the genotype space $\mathcal{G}$, a compatible family for $\mathcal{G}$ with respect to the distribution presheaf $\mathcal{D}_R\mathcal{E}$ is given by a family of distributions $\{d_O \colon P^O \rightarrow R | O \in \mathcal{G}\}$ such that
\begin{eqnarray}
\forall O \in \mathcal{G} \left[ \exists d_O \in \mathcal{D}_R\mathcal{E}(O) \right],\\
d_O|O \cap O' = d_{O'}|O \cap O'.
\end{eqnarray}
The second condition, $d_O|O \cap O' = d_{O'}|O \cap O'$, is referred to as the \emph{sheaf condition}. This condition means that any two distributions $d_O$ and $d_{O'}$ in the \emph{compatible family} of distributions produce the same values in the semiring $R$ on all of the alleles in the intersection of $O$ with $O'$. 

For example, if we have two gene regulatory network modules given by their genotypes $O = \{l_1, l_2\}$ and $O' = \{l_1, l'_2\}$ then the sheaf condition specifies that for $e_{\{l_1\}} \in \mathcal{E}(\{l_1\})$, which assigns a particular phenotype to the genotype $\{l_1\}$, that
\begin{eqnarray}
\sum_{e \in \mathcal{E}(O),\, e|l_1=e_{\{l_1\}}} d_O(e) \,\, = \sum_{e' \in \mathcal{E}(O'),\, e'|l_1=e_{\{l_1\}}} d_{O'}(e')
\end{eqnarray}
This condition means that the probability for gene $i=1$ to be associated to the phenotype given by $e_{\{l_1\}}$ is equivalent in case we do or do not marginalize over possible alleles for gene $i=2$.

\subsection*{Global sections of distributions over genotype-phenotype maps}
To say that the sheaf condition holds for a compatible family $\{d_O\}_{O \in \mathcal{G}}$ for the presheaf $\mathcal{D}_R\mathcal{E}$ implies the existence of a \emph{global section} $d \in \mathcal{D}_R\mathcal{E}(L)$ that is defined on the full set of alleles. This global section defines a distribution on the set of genotype-phenotype maps given by $\mathcal{E}(L) = P^L$, specifying phenotypes associated to all alleles at once instead of on subsets of $L$ associated to either $U \subseteq L$ or, what is essentially equivalent, $O \in \mathcal{G}$. Crucially, the distribution $d$ must also restrict for the subset of alleles associated to any gene regulatory network module, which in some cases may represent all of the genes in an organism, $O$ in a covering of the genotype space $\mathcal{G}$ meaning that
\begin{eqnarray}
\forall O \in \mathcal{G} \left[ d|O = d_O \right].
\end{eqnarray}
In terms of distributions, the existence of a global section $d$ for a covering of the genotype space $\mathcal{G}$ corresponds to the existence of a distribution defined on all alleles that marginalizes to yield the distributions on subsets of alleles specified by the environmental conditions.

The presheaf $\mathcal{E}$ alone was already a sheaf because given a cover $\{U_i\}_{i \in I}$ of $U$ there is a family of sections $\{e_i \in \mathcal{E}(U_i)\}_{i \in I}$ that is compatible in the sense that
\begin{eqnarray}
\forall i,j \in I \left[ e_i|U_i \cap U_j = e_j|U_i \cap U_j \right].
\end{eqnarray}
In this case there is a unique global section $e \in \mathcal{E}(U)$ such that $\forall i \in I \left[ e|U_i = e_i \right]$. The fact that the sheaf condition holds for $\mathcal{E}$ is true because we are considering functions defined on a space with a trivial discrete topology allowing us to combine any partial functions that agree on overlapping subsets by taking the union of the data defining such functions. 

Extending this example on an arbitrary subset $U \subseteq L$ to the whole set of alleles $L$ we have a unique global section $e \in \mathcal{E}(L) = P^L$. $e$ can be used to deterministically assign phenotypes to the universe of alleles $L$ under consideration.

\subsection*{Example}
We can begin to construct an example to make the model description, so far, more concrete. For this example, we take the full set of potential alleles to be $L = \{ l_1,l'_1,l_2,l'_2 \}$. Consider the case in which each of the gene regulatory network modules under consideration each has two genetic loci, $L_1$ and $L_2$, each of which is capable of containing one of two different alleles: $l_1$ or $l'_1$ for $L_1$ and $l_2$ or $l'_2$ for $L_2$. Then we have a covering of the genotype space given by $\mathcal{G} = \{\{l_1,l_2 \},\{l_1,l'_2 \},\{l'_1,l_2\},\{l'_1,l'_2\} \}$. We also have two phenotype values $P = \{0, 1\}$. Each genotype-phenotype map in Table \ref{tab:gpm} assigns phenotype values to each $O \in \mathcal{G}$. For example, $e_1 \colon	l_1 l_2 \rightarrow 00$. Table \ref{tab:probabilities} likewise assigns probabilities to each of these genotype-phenotype mappings.

\input{tex/table_example}

We can collect all of these genotype-phenotype mappings into one set using the functor given in Equation \ref{eq:gpfunctor} as $\mathcal{E}(\mathcal{G}) = P^{\mathcal{G}} = \{e_i | i=1 \ldots 16 \}$. We may further apply the $R$-distribution functor $\mathcal{D}_R$ given in Equation \ref{eq:distfunctor} to obtain a distribution on the genotype-phenotype mappings. Doing so yields  $\mathcal{D}_R\mathcal{E}(\mathcal{G})=d \colon P^\mathcal{G} \rightarrow R$. If we now apply this distribution function $d$ to the set of genotype-phenotype mappings we get the probabilities associated to each genotype-phenotype map $d(\{e_i | i=1 \ldots 16 \}) = \{p_i|i=1 \ldots 16\}$ given in Table \ref{tab:probabilities}. This can also be written more abstractly as $\left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](\{e_i | i=1 \ldots 16 \}) = \left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](P^\mathcal{G}) = \left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](\mathcal{E}(\mathcal{G})) = \{p_i|i=1 \ldots 16\}$. Notice that we applied the functors $\mathcal{E}$ and $\mathcal{D}_R$ to a specific set, which was all of $\mathcal{G}$. We could have applied these functors to any $O \in \mathcal{G}$. In one case, for example if we take $O = \{l_1, l_2\}$, we would have gotten $\mathcal{E}(O) = \{e_i|i=1 \ldots 4\}$ and $\mathcal{D}_R\mathcal{E}(O) = d \colon P^O \rightarrow R$. In this case $\left[\mathcal{D}_R\mathcal{E}(O)\right](\mathcal{E}(O)) = \{p_i|i=1 \ldots 4\}$. Note that this is equivalent to $\left[\mathcal{D}_R\mathcal{E}(\mathcal{G})\right](\mathcal{E}(O))$ since $O \in \mathcal{G}$.

Now if we put $\mathcal{G} = L$ and $O = \{l_1\}$ we may ask for the meaning of $\left[\mathcal{D}_R\mathcal{E}(L)\right](\mathcal{E}(\{l_1\}))$. We find that answering this question requires the introduction of another probability table like Table \ref{tab:il1}. Furthermore, the sheaf condition then introduces a system of equations relating and thus constraining the probability parameters, beyond the constraint that each row sum to one, indicated in like colors between the columns of Table \ref{tab:sheaf} :
\begin{align}\label{eq:pparsys}
p^*_1 &= p_1 + p_2 + p_5 + p_6, &
p^*_2 &= p_3 + p_4 + p_7 + p_8,\\
p^*_3 &= p_9 + p_{10} + p_{13} + p_{14},&
p^*_4 &= p_{11} + p_{10} + p_{12} + p_{13},\\
p^*_5 &= p_1 + p_3 + p_9 + p_{11},&
p^*_6 &= p_2 + p_4 + p_{10} + p_{11},\\
p^*_7 &= p_5 + p_7 + p_{13} + p_{15},&
p^*_8 &= p_6 + p_8 + p_14 + p_16.
\end{align}

\input{tex/table_sheaf}

\subsection*{Checking for the existence of a global section in a constant environment}


\subsection*{Subsection 2}

\section*{Discussion}

% You may title this section "Methods" or "Models". 
% "Models" is not a valid title for PLoS ONE authors. However, PLoS ONE
% authors may use "Analysis" 
\section*{Materials and Methods}

A local section over an open set $U$ of a topological space $X$ is map $s$ such that:

\begin{eqnarray}
s \colon U \rightarrow O
\end{eqnarray}

The set of sections over a particular $U$ can be noted as $O^{U}$. 

A presheaf $F$ associates to each open set $U'$ a a set $O^{U'}$ of sections on $U'$ and to each open set $U \subseteq U'$ a map $F(U') \rightarrow F(U)$ that specifies the restrictions of the sections over $U'$ to $U$.

Sections are locally determined whenever to sections $s_1$ and $s_2$ over $U'$ coincide when they are restricted to $U$. A locally determined presheaf is called separated. A sheaf is a separated presheaf that can be glued together when a unique section $s$  of $U'$ can be restricted to the sections of the family of open sets $U_i$ that form a cover on $U'$. compatible sections can be glued together. They have to  match on the overlaps of the covering set.

Then, under some circumstances, we can glue together all the and then obtain a sheaf. If we are given a family of open sets $U
_i$ such that they form a cover and a compatible family of sections such that:

There is a unique section such that so we can glue together the local data.  



The category theory generalization of the previously described mathematical structure is the presheaf functor. For a category $\mathcal{C}$ the S-valued presheaf F is a functor such that: 
 
$$F \colon \mathcal{C}^{opp} \rightarrow \textit{S}$$.

Where $S$ can be any category like the category of sets ($Set$), commutative rings ($CRing$) or the category of Abelian groups ($Ab$). In order to recover the notion of presheaf in topological spaces, we have to take the $Set$ valued presheaf of the category $\mathcal{C}$ constituted by the posets of open sets a topological space.

$$\xymatrix{
U \ar@{^{(}->}[r] \ar@{^{(}->}[d]
&
U' \ar@{^{(}->}[d]
\\
U'' \ar@{^{(}->}[r]
&
U'''
}
\xymatrix{
 \ar[r]^-{F} &  
}
\xymatrix{
O^{U}  
&
O^{U'}\ar[l]^{res} 
\\
O^{U''}\ar[u] 
&
O^{U'''}\ar[l]\ar[u]
}$$

Then, the presheaf functor takes the open subsets of $\mathcal{C}$ and maps them to set of sections on the open subset $O^{U}$. The morphisms of the $\mathcal{C}^{opp}$ category are indeed retractions and are mapped by the presheaf functor into restrictions of sections: $$res_{U,U'}: F(U') \rightarrow F(U)$$

Such that the composition rules and identities are satisfied in order to constitute a category. The sheaf condition can be expressed also in categorical terms, but we will leave this 

So the question posed at the beginning, can we use this mathematical structure to enrich our perception of what we consider a fitness landscape.
% Do NOT remove this, even if you are not including acknowledgments
\section*{Acknowledgments}


%\section*{References}
\bibliographystyle{plos2009}
% The bibtex filename
\bibliography{template}

\section*{Figure Legends}
%\begin{figure}[!ht]
%\begin{center}
%%\includegraphics[width=4in]{figure_name.2.eps}
%\end{center}
%\caption{
%{\bf Bold the first sentence.}  Rest of figure 2  caption.  Caption 
%should be left justified, as specified by the options to the caption 
%package.
%}
%\label{Figure_label}
%\end{figure}


\section*{Tables}
%\begin{table}[!ht]
%\caption{
%\bf{Table title}}
%\begin{tabular}{|c|c|c|}
%table information
%\end{tabular}
%\begin{flushleft}Table caption
%\end{flushleft}
%\label{tab:label}
% \end{table}

\end{document}

