\input{plos_header.tex}

\begin{document}

% Title must be 150 characters or less
\begin{flushleft}
{\Large
\textbf{Modularization of global genotype-phenotype mappings accesses a relative superset of biological functions}
}
% Insert Author names, affiliations and corresponding author email.
\\
Author1$^{1}$,
Author2$^{2}$,
Author3$^{3,\ast}$
\\
\bf{1} Author1 Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{2} Author2 Dept/Program/Center, Institution Name, City, State, Country
\\
\bf{3} Author3 Dept/Program/Center, Institution Name, City, State, Country
\\
$\ast$ E-mail: Corresponding author@institute.edu
\end{flushleft}

% Please keep the abstract between 250 and 300 words
\section*{Abstract}
\input{tex/abstract}

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLoS ONE authors please skip this step.
% Author Summary not valid for PLoS ONE submissions.
\section*{Author Summary}
\input{tex/authorsummary}

\section*{Introduction}
\input{tex/introduction}

% Results and Discussion can be combined.
% otherwise
% \section*{Discussion}
% \section*{Discussion}
\section*{Results and Discussion}
\subsection*{Coarse-graining gene expression admits a simple probabilistic model}
To motivate our model, we first consider a simple representation of gene expression data. Consider a gene regulatory network (GRN) model defined on a generic representation of a bacterial chromosome or on a given plasmid (Fig. \ref{fig:expression_concept}A). The amount of a given transcript present in a cell given in terms of the discrete counts obtained via sequence census methods (e.g. RNA-seq) or relative abundance derived from microarray data can be binned into a smaller number of discrete classes by setting a collection of thresholds on the original data set. If only a single threshold is given, then the data can be binned into two classes depending upon whether or not the original measurement surpasses the given threshold Fig. \ref{fig:expression_concept}B. In this way, the data derived from a number of samples of a given network can be described using binary sequences Fig. \ref{fig:expression_concept}C. For a given number of genes, it is possible to sample any subset of genes and the structure of this sampling process plays a role in determining the interpretation of the structure of the underlying process that generates the data, which, in this case, includes all regulatory elements that impinge upon the GRN module under consideration. For example, if, whenever we sample, we always consider the expression state of three out of three genes, we imply that the underlying process can be represented as a joint probability distribution over those three genes Fig. \ref{fig:expression_concept}C top. If instead we consider each pair of genes separately Fig. \ref{fig:expression_concept}, we imply the existence of a different kind of structure, to be made precise in what proceeds, within a probability distribution over the collection of genes within the network that has been carved off for measurement.

\subsection*{Divergence between gene expression observational structure and underlying process structure can lead to inconsistency}
We can use an undirected graph (e.g. Fig. \ref{fig:inconsistentthreecycle}A, bottom left) to indicate correlation structures allowed by various GRN architectures (GRNAs). Each node of the graph is associated to the probability distribution that specifies probabilities for each gene to be observed in each of the states specified by the coarse-graining process described in the previous section. Each edge of the graph specifies a joint probability distribution for both of the nodes it contains (or connects) to simultaneously take on a given pair of values. In this context it is presumed that via marginalization the joint probability distributions associated to each edge are consistent with the probability distributions associated to each node.

Fig. \ref{fig:inconsistentthreecycle}A provides a concrete example where we assume three genes interact via all possible pairwise combinations and that via the coarse-graining process we have binned the expression state of each gene into one of two classes. Note that this graphical model is completely general and we can have, in principle, an arbitrary number of genes and an arbitrary number of classes in the assignment of expression states up to the maximal resolution of the method by which the observations are performed. Each node of the graph in Fig. \ref{fig:expression_concept}A represents a probability distribution over the observation of each gene in either of the two states established in the coarse-graining process, which for node three are given by $p((3 \mapsto 0, 3 \mapsto 1))=(0.5,0.5)$. Given an edge we consider a map or function that takes as input the two genes connected by or contained within that edge and maps those two genes simultaneously to the available expression states. For example in Fig. \ref{fig:expression_concept}A bottom there are maps $12 \mapsto 11$, $23 \mapsto 10$, and $31 \mapsto 10$. Each of the probability tables adjacent to each edge in the graph in Fig. \ref{fig:expression_concept}A top assigns a probability distribution to a set of the maps from the nodes connected by the edge to all possible combinations of the expression states. As these maps take collections of genes as input and produce collections of expression states as outputs we refer to them as genotype-phenotype mappings and the associated probability distributions are thus probability distributions over genotype-phenotype maps.

If the probability distribution in Fig. \ref{fig:inconsistentthreecycle}A represents the actual structure and particular parameters of the process generating the gene expression data, there is a dual view of this graph in which we consider the inverse problem of inferring the parameters of the probability distribution specified adjacent to the nodes and edges of the graph given the assumption that the structure of a given GRN module takes on the architecture of the given graph and the observation of data according to that assumption. Fig. \ref{fig:inconsistentthreecycle}B represents data generated by a process whose structure is equivalent to that of Fig \ref{fig:inconsistentthreecycle}A except that the parameters of the probability distributions are such that each of the distributions is uniform. Fig \ref{fig:inconsistentthreecycle}B left represents a sample in each row for each of genes one, two and three while the rows are placed in a random order. Fig \ref{fig:inconsistentthreecycle}B right shows that indeed all possible binary sequences representing the expression states for genes one, two and three are approximately uniformly represented in the given samples. We now use a standard inference algorithm to infer the parameters of a model like that of Fig. \ref{fig:inconsistentthreecycle}A given the dataset represented in Fig. \ref{fig:expression_concept}C bottom, which is consistent with the specific values of the parameters given in the probability tables adjacent to the nodes and edges in Fig. \ref{fig:inconsistentthreecycle}A. If we compute the empirical distribution by simply normalizing the data in Fig. \ref{fig:expression_concept}C bottom and compare it to the data generated by the trained model given in Fig. \ref{fig:inconsistentthreecycle}C, we find that only the data of Fig. \ref{fig:expression_concept}C bottom is consistent with the probabilities represented in Fig. \ref{fig:inconsistentthreecycle}A.

This inconsistency can be viewed as a result of the fact that the network architecture in Fig. \ref{fig:inconsistentthreecycle}A contains a cycle and that we have chosen a set of parameters that leads to the impossibility of finding a joint probability distribution over all three genes. For the case of the architecture in Fig. \ref{fig:inconsistentthreecycle}A, and moreover for any GRNA of any size that contains one or more cycles, the possibility of finding a joint distribution over all genes requires the implicit assumption that the structure of the generating process can be viewed simultaneously as that of Fig. \ref{fig:expression_concept}C top and that of Fig. \ref{fig:expression_concept}C bottom. This means that if we consider all possible parameters, the space of probability distributions associated to the two GRNAs contrasted in Fig. \ref{fig:expression_concept}C are different. This recognition leads to the question of whether or not it is possible to classify the geometries and associated relationships among the spaces of probability distributions associated to all possible GRNAs.

\subsection*{Gene-regulatory network architecture is associated to the geometry of spaces of probability distributions over genotype-phenotype maps}
The collection of all possible GRNAs in the context of graphical models is equivalent to the lattice of all possible subsets of genes. For example, Fig. \ref{fig:conediagram}A contains the lattice of subsets of three genes. We are only interested in those subsets that contain at least one copy of each gene, which corresponds to the region highlighted with a grey background in Fig. \ref{fig:conediagram}A. Each GRNA induces a different structure on the genotype-phenotype maps underlying it. For example, Fig. \ref{fig:conediagram}B shows in identical vertical order the different structures induced by the three architectures highlighted in green in Fig. \ref{fig:conediagram}A.

We consider those GRNAs found lower in the lattice of Fig. \ref{fig:conediagram}A to be more highly constrained or modular because of the way in which each corresponds to the increasing restriction of simultaneous evaluation of phenotypes over the collection of genes. Fig. \ref{fig:conediagram}B top corresponds to the least modular GRNA because phenotype values are assigned simultaneously to all three genes. Fig. \ref{fig:conediagram}B middle exhibits an elevated degree of modularity because phenotypes are only simultaneously assigned to pairs of genes despite the fact that on timescales longer than that fundamental to the generating process it is possible to measure the phenotype values of all three genes simultaneously. Similarly, Fig. \ref{fig:conediagram}B bottom is even more modular because phenotype values are only assigned to one gene at a time.

Each of the GRNAs in Fig. \ref{fig:conediagram}A can be associated to a space of probability distributions over genotype-phenotype mappings. Fig. \ref{fig:conediagram}C schematically depicts the relationships among the probability distributions associated to the corresponding architectures and genotype-phenotype mappings in Fig. \ref{fig:conediagram}B. The inconsistency noted in the previous section between the architectures Fig. \ref{fig:conediagram}B top and middle is a result of the differing geometries in Fig. \ref{fig:conediagram}C middle. There, the smaller darker grey region corresponds to the space of all possible probability distributions for all possible parameters achievable via application of the marginalization map to the probability distribution defined over all possible genotype-phenotype mappings associated to the network architecture in Fig. \ref{fig:conediagram}B top. Similarly, the lighter grey region corresponds to the analogous structure for Fig. \ref{fig:conediagram}B middle.

This relationship indicates that there are certain cases in which a given GRNA exhibiting a relatively higher degree of modularity nevertheless provides access to a larger space of possible probability distributions and thus of correlations among genes which can be coarsely referred to as gene expression patterns. These patterns serve as substrate for selection of GRNAs with particular functions by the network context within which the small modules we have considered so far function.

\subsection*{Some GRNAs provide access to larger collections of gene expression correlations than others}
Relationships between spaces of potential gene expression patterns like that of Fig. \ref{fig:conediagram}C middle occur for all GRNAs on any number of genes so long as there exists at least one cycle in the corresponding network architecture. For the case of three genes, there is only one class of graphs containing a cycle, which is that of Fig. \ref{fig:conediagram}B middle. For the case of four genes there are nine different classes of graphs containing cycles and these nine classes can be split into two groups depending upon whether or not the edges of the graphs are each restricted to contain only two genes. Fig. \ref{fig:non2uniformcyclichypergraphhasse} shows the componenets of the lattice analogous to that of Fig. \ref{fig:conediagram}A containing the five out of the nine graphs on four genes having cycles contained in the subclass allowing any number of genes to be contained in an edge of the graph.

Given the larger collection of GRNAs with cycles we can assess the relative sizes of the analogs to the light and dark grey spaces (Fig. \ref{fig:conediagram}C middle) of probability distributions over genotype-phenotype maps. We assess the likelihood of choosing a point in the dark grey region at random by computing the ratio of the volume of the dark grey region (associated to the non-modular GRNA analogous to that of Fig. \ref{fig:conediagram}B top with a single edge containing all four genes), whose architecture and thus volume is fixed, to that of the light grey region, whose volume varies according to each of the cyclic graphs associated to a GRNA on four genes. We refer to this number as the non-modular:modular volume ratio. Fig. \ref{fig:ncycvolrat}A shows the relevant computations for seven different graphs where the edges are restricted to each contain two genes. The spaces are equivalent and thus the volume ratio equal to one for graphs lacking cycles (e.g. the first three graphs along the x-axis of Fig. \ref{fig:ncycvolrat}A). For the four network architectures in Fig. \ref{fig:ncycvolrat}A containing cycles, we see that the volume ratio is consistently less than one. This quantifies the degree to which the space of probability distributions associated to the network architecture depicted along the x-axis is larger than the space of probability distributions associated to the non-modular architecture.

\subsection*{Network context can impose constraints on network architecture via the requirement of particular expression patterns over evolutionary timescales}

\subsection*{Discussion}
The goal of the measurement of gene expression has to be to uncover the structure of the generative process encoded in the GRN interactions involved. This fact has motivated the development of a large collection of algorithms to infer aspects of this structure \cite{DeSmet2010} and to compare networks \cite{Ideker2012}. Our model highlights that one of the most important factors in improving the relatively low fidelity of this process is to consider the space of possible correlations imposed by a given measurement structure.

One important challenge for the applicability of the tools surrounding graphical models and their associated probability distributions is to establish a precise mapping between the representation of GRNs in terms of promoter diagrams and the structure of Markov random fields or Bayesian networks.

% This section is often entitled "Materials and Methods"
% You may title this section "Methods" or "Models".
% "Models" is not a valid title for PLoS ONE authors. However, PLoS ONE
% authors may use "Analysis"
\section*{Models}

\input{tex/modeldescription}
\input{tex/marginalmapconstraints}
%\input{tex/hadamardmethod}

% Do NOT remove this, even if you are not including acknowledgments
\section*{Acknowledgments}


%\section*{References}
% The bibtex filename
\bibliography{bib/books,bib/papers}

\section*{Figure Legends}
%\begin{figure}[!ht]
%\begin{center}
%%\includegraphics[width=4in]{figure_name.2.eps}
%\end{center}
%\caption{
%{\bf Bold the first sentence.}  Rest of figure 2  caption.  Caption
%should be left justified, as specified by the options to the caption
%package.
%}
%\label{Figure_label}
%\end{figure}
\input{tex/figures}

\FloatBarrier

\section*{Tables}
%\begin{table}[!ht]
%\caption{
%\bf{Table title}}
%\begin{tabular}{|c|c|c|}
%table information
%\end{tabular}
%\begin{flushleft}Table caption
%\end{flushleft}
%\label{tab:label}
% \end{table}
\input{tex/table_hidvarmod}
\input{tex/table_example}
\input{tex/table_sheaf}
\input{tex/table_pars222}
\input{tex/table_logmat222}

\FloatBarrier

\section*{Polymake computation of the non-modular:modular volume ratio of probability distribution polytopes for cyclic hypergraphs on three and four vertices}
\input{tex/hypergraphvolrat}
\pagebreak

\section*{Example graphs on four vertices}
\input{tex/graphlist4}
\pagebreak

\section*{Generate the lattice of hypergraphs on a given number of vertices}
\input{tex/hypergraphlattice}
\pagebreak

\section*{Filter hypergraphs with cycles using Graham reduction}
\input{tex/hypergraphcycle}
\pagebreak

\section*{Approximate volume computations}
\input{tex/overfourapproxvol}

\end{document}
